From 04d03d060c57c388545501ed588d8aeff5d744b2 Mon Sep 17 00:00:00 2001
From: JC Fillion-Robin <jcfr@nvidia.com>
Date: Thu, 23 Apr 2026 18:58:41 -0400
Subject: [PATCH] Updates for OE cross builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 584c041..5a22196 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,32 @@
 # Minimum CMake required
 cmake_minimum_required(VERSION 3.26)
+
+function(onnx_extract_version)
+    file(READ "${CMAKE_SOURCE_DIR}/VERSION_NUMBER" ONNX_VERSION)
+    string(STRIP "${ONNX_VERSION}" ONNX_VERSION)
+    string(REGEX MATCH "^([0-9]+)\\.([0-9]+)\\.([0-9]+)" ONNX_VERSION_MATCH ${ONNX_VERSION})
+    message(STATUS "ONNX matched version: ${ONNX_VERSION_MATCH}")
+    if(ONNX_VERSION_MATCH)
+        # Extract the major version
+        string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" ver_major ${ONNX_VERSION})
+        # Extract the minor version
+        string(REGEX REPLACE "^[0-9]+\\.([0-9]+)\\..*" "\\1" ver_minor ${ONNX_VERSION})
+        # Extract the patch version
+        string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]+)" "\\1" ver_patch ${ONNX_VERSION})
+
+        set(ONNX_VERSION_MAJOR ${ver_major} PARENT_SCOPE)
+        set(ONNX_VERSION_MINOR ${ver_minor} PARENT_SCOPE)
+        set(ONNX_VERSION_PATCH ${ver_patch} PARENT_SCOPE)
+        set(ONNX_VERSION "${ver_major}.${ver_minor}.${ver_patch}" PARENT_SCOPE)
+    else()
+        message(FATAL_ERROR "ONNX version did not match the expected format.")
+    endif()
+endfunction()
+
+onnx_extract_version()
+
 # Project
-project(onnx LANGUAGES CXX)
+project(onnx VERSION ${ONNX_VERSION} LANGUAGES CXX)
 
 include(cmake/Utils.cmake)
 # Honor visibility properties for all target types.
@@ -413,10 +438,12 @@ if(ONNX_USE_LITE_PROTO)
   endif()
 endif()
 add_library(onnx_proto ${ONNX_PROTO_SRCS})
+set_target_properties(onnx_proto PROPERTIES VERSION ${ONNX_VERSION} SOVERSION ${ONNX_VERSION_MAJOR}.${ONNX_VERSION_MINOR})
 add_onnx_global_defines(onnx_proto)
 
 add_library(onnx ${ONNX_SRCS} ${ONNX_PROTO_SRCS})
 add_dependencies(onnx onnx_proto)
+set_target_properties(onnx PROPERTIES VERSION ${ONNX_VERSION} SOVERSION ${ONNX_VERSION_MAJOR}.${ONNX_VERSION_MINOR})
 set_source_files_properties(${ONNX_SRCS} PROPERTIES CXX_VISIBILITY_PRESET hidden)
 add_onnx_global_defines(onnx)
 
