From 3865b7dbb64524cdfcf45ab2b9b18c4112412f39 Mon Sep 17 00:00:00 2001
From: Julien Jomier <jjomier@nvidia.com>
Date: Thu, 23 Apr 2026 17:39:48 -0400
Subject: [PATCH] Remove cross-compile CUDA test

---
 test/apps/Makefile.am | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/test/apps/Makefile.am b/test/apps/Makefile.am
index dbd98ba..668cde2 100644
--- a/test/apps/Makefile.am
+++ b/test/apps/Makefile.am
@@ -98,23 +98,6 @@ test_cuda_hook_static_LDADD     = $(top_builddir)/src/ucp/libucp.la \
                                   $(CUDART_STATIC_LIBS) $(CUDA_LIBS)
 endif
 
-if HAVE_NVCC
-
-include $(top_srcdir)/config/cuda.am
-
-# Could be simply added as .cu to _SOURCES if there was no specific CPPFLAGS
-$(eval $(call nvcc-source, custom-test_cuda_get_symbol_address.o, \
-	test_cuda_get_symbol_address.cu,CUDA_CPPFLAGS))
-
-noinst_PROGRAMS                      += test_cuda_get_symbol_address
-test_cuda_get_symbol_address_SOURCES  =
-test_cuda_get_symbol_address_LDFLAGS  = $(CUDA_LDFLAGS)
-test_cuda_get_symbol_address_LDADD    = $(CUDART_LIBS) custom-test_cuda_get_symbol_address.o \
-                                        $(top_builddir)/src/ucp/libucp.la \
-                                        $(top_builddir)/src/ucm/libucm.la
-
-endif
-
 endif
 
 if HAVE_TCMALLOC
-- 
2.50.1

