diff -urN tegra-nvfancontrol-35.3.1-20230319081403.orig/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_b01.conf tegra-nvfancontrol-35.3.1-20230319081403/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_b01.conf
--- tegra-nvfancontrol-35.3.1-20230319081403.orig/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_b01.conf	1969-12-31 17:00:00.000000000 -0700
+++ tegra-nvfancontrol-35.3.1-20230319081403/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_b01.conf	2023-05-16 13:31:41.821806363 -0600
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+POLLING_INTERVAL 2
+
+# tegra module side fan
+<FAN 2>
+	TMARGIN ENABLED
+	FAN_GOVERNOR pid {
+		STEP_SIZE 10
+	}
+	FAN_GOVERNOR cont {
+		STEP_SIZE 10
+	}
+	FAN_CONTROL close_loop {
+		RPM_TOLERANCE 100
+	}
+	FAN_PROFILE quiet {
+		#TEMP 	HYST	PWM	RPM
+		0	0 	255 	2700
+		5	0 	255	2700
+		20	0	105	1800
+		35	0	40	1400
+		70	0	0	1000
+		115	0	0	1000
+	}
+	THERMAL_GROUP 0 {
+		GROUP_MAX_TEMP 115
+		#Thermal-Zone Coeffs Max-Temp
+		CPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		GPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC0-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC1-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC2-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+	}
+	FAN_DEFAULT_CONTROL close_loop
+	FAN_DEFAULT_PROFILE quiet
+	FAN_DEFAULT_GOVERNOR cont
+	KICKSTART_PWM 40
+
+# dGPU side fan
+<FAN 1>
+	TMARGIN ENABLED
+	FAN_GOVERNOR pid {
+		STEP_SIZE 10
+	}
+	FAN_GOVERNOR cont {
+		STEP_SIZE 10
+	}
+	FAN_CONTROL close_loop {
+		RPM_TOLERANCE 100
+	}
+	FAN_PROFILE quiet {
+		#TEMP 	HYST	PWM	RPM
+		0	0 	255 	2700
+		5	0 	255	2700
+		20	0	105	1800
+		35	0	40	1400
+		70	0	0	1000
+		115	0	0	1000
+	}
+	THERMAL_GROUP 0 {
+		GROUP_MAX_TEMP 115
+		#Thermal-Zone Coeffs Max-Temp
+		CPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		GPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC0-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC1-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC2-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+	}
+	FAN_DEFAULT_CONTROL close_loop
+	FAN_DEFAULT_PROFILE quiet
+	FAN_DEFAULT_GOVERNOR cont
+	KICKSTART_PWM 40
diff -urN tegra-nvfancontrol-35.3.1-20230319081403.orig/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_c01.conf tegra-nvfancontrol-35.3.1-20230319081403/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_c01.conf
--- tegra-nvfancontrol-35.3.1-20230319081403.orig/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_c01.conf	1969-12-31 17:00:00.000000000 -0700
+++ tegra-nvfancontrol-35.3.1-20230319081403/etc/nvpower/nvfancontrol/nvfancontrol_p3740_0002_c01.conf	2023-05-16 13:31:41.821806363 -0600
@@ -0,0 +1,81 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+POLLING_INTERVAL 2
+
+# tegra module side fan
+<FAN 2>
+	TMARGIN ENABLED
+	FAN_GOVERNOR pid {
+		STEP_SIZE 10
+	}
+	FAN_GOVERNOR cont {
+		STEP_SIZE 10
+	}
+	FAN_CONTROL close_loop {
+		RPM_TOLERANCE 100
+	}
+	FAN_PROFILE quiet {
+		#TEMP 	HYST	PWM	RPM
+		0	0 	255 	2700
+		5	0 	255	2700
+		20	0	105	1800
+		35	0	40	1400
+		70	0	0	1000
+		115	0	0	1000
+	}
+	THERMAL_GROUP 0 {
+		GROUP_MAX_TEMP 115
+		#Thermal-Zone Coeffs Max-Temp
+		CPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		GPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC0-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC1-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC2-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+	}
+	FAN_DEFAULT_CONTROL close_loop
+	FAN_DEFAULT_PROFILE quiet
+	FAN_DEFAULT_GOVERNOR cont
+	KICKSTART_PWM 40
+
+# dGPU side fan
+<FAN 4>
+	TMARGIN ENABLED
+	FAN_GOVERNOR pid {
+		STEP_SIZE 10
+	}
+	FAN_GOVERNOR cont {
+		STEP_SIZE 10
+	}
+	FAN_CONTROL close_loop {
+		RPM_TOLERANCE 100
+	}
+	FAN_PROFILE quiet {
+		#TEMP 	HYST	PWM	RPM
+		0	0 	255 	2700
+		5	0 	255	2700
+		20	0	105	1800
+		35	0	40	1400
+		70	0	0	1000
+		115	0	0	1000
+	}
+	THERMAL_GROUP 0 {
+		GROUP_MAX_TEMP 115
+		#Thermal-Zone Coeffs Max-Temp
+		CPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		GPU-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC0-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC1-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+		SOC2-therm 20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 0
+	}
+	FAN_DEFAULT_CONTROL close_loop
+	FAN_DEFAULT_PROFILE quiet
+	FAN_DEFAULT_GOVERNOR cont
+	KICKSTART_PWM 40
