From d9d83a157d03f0bdd8049c6de62470c715e5c5d1 Mon Sep 17 00:00:00 2001
From: Soham Sinha <sohams@nvidia.com>
Date: Wed, 22 Feb 2023 20:06:45 +0000
Subject: [PATCH] Add MMU_NOTIFIER dependency in nv-p2p-Kconfig

---
 nvidia/drivers/nv-p2p/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nvidia/drivers/nv-p2p/Kconfig b/nvidia/drivers/nv-p2p/Kconfig
index c28b3dd6f8c3..653a9b5be1d2 100644
--- a/nvidia/drivers/nv-p2p/Kconfig
+++ b/nvidia/drivers/nv-p2p/Kconfig
@@ -5,5 +5,6 @@
 menuconfig TEGRA_RDMA
 	tristate "TEGRA_RDMA_DRIVER"
 	default n
+	depends on MMU_NOTIFIER
 	help
 	    If in doubt, say N.
-- 
2.25.1

