diff -urN a/nvidia/drivers/media/platform/tegra/camera/vi/vi5_formats.h b/nvidia/drivers/media/platform/tegra/camera/vi/vi5_formats.h
--- a/nvidia/drivers/media/platform/tegra/camera/vi/vi5_formats.h	2023-06-09 15:26:38.840628131 -0600
+++ b/nvidia/drivers/media/platform/tegra/camera/vi/vi5_formats.h	2023-06-09 15:27:17.061069111 -0600
@@ -118,7 +118,7 @@
 
 	/* RGB888 */
 	TEGRA_VIDEO_FORMAT(RGB888, 24, RGB888_1X24, 4, 1, T_A8R8G8B8,
-				RGB888, ABGR32, "BGRA-8-8-8-8"),
+				RGB888, RGBA32, "RGBA-8-8-8-8"),
 	TEGRA_VIDEO_FORMAT(RGB888, 24, RGB888_1X32_PADHI, 4, 1, T_A8B8G8R8,
 				RGB888, RGB32, "RGB-8-8-8-8"),
 
