From c5d2582811c335b0ee62747a00d9c97b9c3f2f99 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 29 Mar 2023 14:21:37 -0600
Subject: prometheus: dts: support large BAR1

---
 .../kernel-dts/cvb/tegra234-p3740-b01-pcie.dtsi        | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/nvidia/platform/t23x/prometheus/kernel-dts/cvb/tegra234-p3740-b01-pcie.dtsi b/nvidia/platform/t23x/prometheus/kernel-dts/cvb/tegra234-p3740-b01-pcie.dtsi
index a4da25535936..7a272b3fa442 100644
--- a/nvidia/platform/t23x/prometheus/kernel-dts/cvb/tegra234-p3740-b01-pcie.dtsi
+++ b/nvidia/platform/t23x/prometheus/kernel-dts/cvb/tegra234-p3740-b01-pcie.dtsi
@@ -34,6 +34,16 @@ pcie@14160000 {
 	pcie@141a0000 {
 		status = "okay";
 
+		reg = <0x00 0x141a0000 0x0 0x00020000   /* appl registers (128K)      */
+		       0x00 0x3a000000 0x0 0x00040000   /* configuration space (256K) */
+		       0x00 0x3a040000 0x0 0x00040000   /* iATU_DMA reg space (256K)  */
+		       0x00 0x3a080000 0x0 0x00040000   /* DBI reg space (256K)       */
+		       0x2e 0x20000000 0x0 0x10000000>; /* ECAM (256MB)               */
+
+		ranges = <0x81000000 0x00 0x3a100000 0x00 0x3a100000 0x0 0x00100000      /* downstream I/O (1MB) */
+		          0x82000000 0x00 0x40000000 0x2e 0x30000000 0x0 0x08000000      /* non-prefetchable memory (128MB) */
+		          0xc3000000 0x27 0x40000000 0x27 0x40000000 0x6 0xe0000000>;    /* prefetchable memory (28160MB) */
+
 		phys = <&p2u_nvhs_0>, <&p2u_nvhs_1>, <&p2u_nvhs_2>,
 		       <&p2u_nvhs_3>, <&p2u_nvhs_4>, <&p2u_nvhs_5>,
 		       <&p2u_nvhs_6>, <&p2u_nvhs_7>;
-- 
2.34.1

