From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Fri, 19 Jan 2024 09:41:47 -0700
Subject: Fix nvfuser install path


diff --git a/third_party/nvfuser/CMakeLists.txt b/third_party/nvfuser/CMakeLists.txt
index c355e185..803b5616 100644
--- a/third_party/nvfuser/CMakeLists.txt
+++ b/third_party/nvfuser/CMakeLists.txt
@@ -18,7 +18,7 @@ file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/nvfuser")
 set(NVFUSER_ROOT ${PROJECT_SOURCE_DIR})
 set(NVFUSER_SRCS_DIR "${NVFUSER_ROOT}/csrc")
 set(TORCH_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/../..")
-set(TORCH_INSTALL_LIB_DIR ${TORCH_ROOT}/torch/lib)
+set(TORCH_INSTALL_LIB_DIR lib)
 
 # --- build nvfuser_codegen library
 
@@ -159,10 +159,6 @@ target_include_directories(${NVFUSER_CODEGEN}
                            PUBLIC $<BUILD_INTERFACE:${NVFUSER_SRCS_DIR}>)
 set_property(TARGET ${NVFUSER_CODEGEN} PROPERTY CXX_STANDARD 17)
 install(TARGETS ${NVFUSER_CODEGEN} EXPORT NvfuserTargets DESTINATION "${TORCH_INSTALL_LIB_DIR}")
-# installing nvfuser python tests
-install(DIRECTORY "${NVFUSER_ROOT}/python_tests/"
-        DESTINATION "${TORCH_ROOT}/test/_nvfuser"
-        FILES_MATCHING PATTERN "*.py" )
 
 file(WRITE "${TORCH_ROOT}/test/_nvfuser/.gitignore" "*")
 # --- build nvfuser_python library
-- 
2.34.1

