From 1086bc7574a83c6e8642a87e7a7531bd5897dafa Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 25 Jan 2023 16:50:14 -0700
Subject: Add nvidia-modprobe-utils static library

---
 Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Makefile b/Makefile
index 6391ca9..3d4566d 100644
--- a/Makefile
+++ b/Makefile
@@ -160,3 +160,15 @@ $(MANPAGE_not_gzipped): nvidia-modprobe.1.m4 $(OPTIONS_1_INC) $(VERSION_MK)
 
 $(MANPAGE_gzipped): $(MANPAGE_not_gzipped)
 	$(GZIP_CMD) -9nf < $< > $@
+
+##### nvidia-modprobe-utils static library #####
+
+UTILS_LIB_STATIC := modprobe-utils/libnvidia-modprobe-utils.a
+UTILS_LIB_INCS   := modprobe-utils/nvidia-modprobe-utils.h \
+                    modprobe-utils/pci-enum.h
+UTILS_LIB_SRCS   := modprobe-utils/nvidia-modprobe-utils.c \
+                    modprobe-utils/pci-sysfs.c
+UTILS_LIB_OBJS   := $(UTILS_LIB_SRCS:.c=.o)
+
+$(UTILS_LIB_STATIC): $(UTILS_LIB_OBJS)
+	$(AR) rs $@ $^
-- 
2.34.1

