From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 24 Jan 2024 15:39:45 -0700
Subject: Fix CMake config dependencies


diff --git a/cmake/gRPCConfig.cmake.in b/cmake/gRPCConfig.cmake.in
index 98d8c6d8b9..ef066858f1 100644
--- a/cmake/gRPCConfig.cmake.in
+++ b/cmake/gRPCConfig.cmake.in
@@ -1,6 +1,8 @@
 # Module path
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/modules)
 
+include(CMakeFindDependencyMacro)
+
 # Depend packages
 @_gRPC_FIND_ZLIB@
 @_gRPC_FIND_PROTOBUF@
-- 
2.34.1

