From aebff6915741a4ca5d88fedc002fae11f70150be Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Tue, 29 Mar 2022 11:43:01 -0600
Subject: [PATCH 1/6] Fix parameter_storage build error

---
 gxf/std/parameter_storage.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gxf/std/parameter_storage.hpp b/gxf/std/parameter_storage.hpp
index 8716e88..689792b 100644
--- a/gxf/std/parameter_storage.hpp
+++ b/gxf/std/parameter_storage.hpp
@@ -22,6 +22,7 @@
 #include <shared_mutex>  // NOLINT
 #include <string>
 #include <utility>
+#include <mutex>
 
 #include "gxf/core/parameter.hpp"
 
-- 
2.17.1

