From b4791ea4ec2acd2033d892ae4c1252b5d542cd21 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 31 Aug 2022 09:23:38 -0600
Subject: Remove TypenameAsString from nvidia namespace


diff --git a/common/type_name.hpp b/common/type_name.hpp
index 49b3f7f..94e0b63 100644
--- a/common/type_name.hpp
+++ b/common/type_name.hpp
@@ -27,8 +27,6 @@
 // Not yet implemented
 #endif
 
-namespace nvidia {
-
 // Gives a string representation of the name of a C++ type.
 //
 // The function will compute the typename during the first invocation and store it in heap memory.
@@ -52,11 +50,9 @@ const char* TypenameAsString() {
   static char s_name[kMaxNameLength] = {0};  // Initialize with 0 to get a null-terminated string.
   static char* result = s_name;
   if (s_name[0] == 0 && result != nullptr) {  // Check for first invokation of this function.
-    result = TypenameAsStringImpl(__PRETTY_FUNCTION__, s_name, kMaxNameLength);
+    result = nvidia::TypenameAsStringImpl(__PRETTY_FUNCTION__, s_name, kMaxNameLength);
   }
   return result;
 }
 
-}  // namespace nvidia
-
 #endif  // NVIDIA_COMMON_TYPE_NAME_HPP_
-- 
2.34.1

