From aa8ec7bfe1a38e825479037bccc60ffa15f88ae5 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 16 Nov 2022 13:36:46 -0700
Subject: Fix GxfEntityCreateInfo initializer


diff --git a/gxf/core/entity.hpp b/gxf/core/entity.hpp
index d326f00..ac24c7b 100644
--- a/gxf/core/entity.hpp
+++ b/gxf/core/entity.hpp
@@ -37,7 +37,7 @@ class Entity {
   // Creates a new entity
   static Expected<Entity> New(gxf_context_t context) {
     gxf_uid_t eid;
-    const GxfEntityCreateInfo info = {0};
+    const GxfEntityCreateInfo info = {0, 0};
     const gxf_result_t code = GxfCreateEntity(context, &info, &eid);
     if (code != GXF_SUCCESS) {
       return Unexpected{code};
-- 
2.34.1

