From 07149cf22133d5bd014cfd1230fe507814bcbf25 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 16 Nov 2022 13:04:33 -0700
Subject: Remove complex primitives support


diff --git a/gxf/std/tensor.hpp b/gxf/std/tensor.hpp
index 9e0712f..2cd9d8f 100644
--- a/gxf/std/tensor.hpp
+++ b/gxf/std/tensor.hpp
@@ -27,7 +27,6 @@
 #include "gxf/core/component.hpp"
 #include "gxf/core/expected.hpp"
 #include "gxf/std/allocator.hpp"
-#include "gxf/std/complex.hpp"
 #include "gxf/std/memory_buffer.hpp"
 
 namespace nvidia {
@@ -46,8 +45,6 @@ enum class PrimitiveType : int32_t {
   kUnsigned64,
   kFloat32,
   kFloat64,
-  kComplex64,
-  kComplex128,
 };
 
 // Returns the size of each element of specific PrimitiveType as number of bytes.
@@ -73,8 +70,6 @@ GXF_PRIMITIVE_TYPE_TRAITS(int64_t, kInt64);
 GXF_PRIMITIVE_TYPE_TRAITS(uint64_t, kUnsigned64);
 GXF_PRIMITIVE_TYPE_TRAITS(float, kFloat32);
 GXF_PRIMITIVE_TYPE_TRAITS(double, kFloat64);
-GXF_PRIMITIVE_TYPE_TRAITS(complex64, kComplex64);
-GXF_PRIMITIVE_TYPE_TRAITS(complex128, kComplex128);
 
 // Type to hold the shape of a tensor
 class Shape {
-- 
2.34.1

