From b218a5bed20529614406704d62dd21c5bee5cde6 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 20 Jul 2023 15:33:04 -0600
Subject: Remove host paths from build


diff --git a/configure.ac b/configure.ac
index 1c0d458..1bfdca3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,8 +90,6 @@ fi
 AM_CONDITIONAL(KERNEL_BUILD,[test ${KERNEL_BUILD_CHK} = 1])
 AC_SUBST(KERNEL_BUILD)
 
-LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/usr/local/lib64 -L/usr/lib64"
-CXXFLAGS="${CXXFLAGS} -isystem /usr/local/include" 
 if test "x$OS" = "xFreeBSD"; then
     AC_MSG_NOTICE(FreeBSD Build)
     MTCR_CONF_DIR="mtcr_freebsd"
-- 
2.34.1

