From db657bb6d58d94e6c0fb42df6471f18e0b017b9f Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 20 Jul 2023 16:35:13 -0600
Subject: Fix include and library build paths


diff --git a/adb_parser/Makefile.am b/adb_parser/Makefile.am
index 0a6891a..734cefb 100644
--- a/adb_parser/Makefile.am
+++ b/adb_parser/Makefile.am
@@ -36,7 +36,7 @@
 
 DIST_SUBDIRS = .
 
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 INCLUDES = -I$(USER_DIR) -I$(USER_DIR)/common
 AM_CXXFLAGS = -Wall -W -g -MP -MD  -pipe $(COMPILER_FPIC)
 
diff --git a/cmdif/Makefile.am b/cmdif/Makefile.am
index b6c43d5..c7d62d3 100644
--- a/cmdif/Makefile.am
+++ b/cmdif/Makefile.am
@@ -35,7 +35,7 @@
 USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/include/mtcr_ul
 TOOLS_LAYOUTS_DIR = $(USER_DIR)/tools_layouts
-AM_CPPFLAGS = -I. -I../common -I../tools_layouts  -I${MTCR_DIR} -I.. -I$(USER_DIR)/${MTCR_CONF_DIR}
+AM_CPPFLAGS = -I. -I$(USER_DIR)/common -I$(USER_DIR)/tools_layouts  -I${MTCR_DIR} -I$(USER_DIR) -I$(USER_DIR)/${MTCR_CONF_DIR}
 CCMDIF_SO = ccmdif.so
 
 AM_CFLAGS = -W -Wall  -g -MP -MD $(COMPILER_FPIC) -DCMDIF_EXPORTS
@@ -51,7 +51,7 @@ libcmdif_a_SOURCES = tools_cif.c tools_cif.h icmd_cif_common.c icmd_cif_common.h
 cmdif_pylibdir = $(libdir)/mstflint/python_tools/
 cmdif_pylib_DATA = ${CCMDIF_SO} cmdif.py
 ${CCMDIF_SO}: libcmdif.a
-	$(CC) -g -Wall -pthread -shared ${CFLAGS} *.o -o ${CCMDIF_SO} \
-	-L$(USER_DIR)/tools_layouts -ltools_layouts -L$(USER_DIR)/${MTCR_CONF_DIR} -lmtcr_ul
+	$(CC) -g -Wall -pthread -shared ${CFLAGS} libcmdif.a -o ${CCMDIF_SO} \
+	-L$(top_builddir)/tools_layouts -ltools_layouts -L$(top_builddir)/${MTCR_CONF_DIR} -lmtcr_ul
 
 CLEANFILES = ${CCMDIF_SO}
diff --git a/cmdparser/Makefile.am b/cmdparser/Makefile.am
index b5367dc..d68e5d3 100644
--- a/cmdparser/Makefile.am
+++ b/cmdparser/Makefile.am
@@ -32,7 +32,7 @@
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
 
-INCLUDES = -I. -I../common
+INCLUDES = -I. -I$(top_srcdir)/common
 
 AM_CPPFLAGS = -W  -g -MP -MD $(COMPILER_FPIC)  $(COMPILER_FPIE)
 
diff --git a/dev_mgt/Makefile.am b/dev_mgt/Makefile.am
index 2283c2d..2df5d5a 100644
--- a/dev_mgt/Makefile.am
+++ b/dev_mgt/Makefile.am
@@ -31,8 +31,8 @@
 #--
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
-USER_DIR = ..
-AM_CPPFLAGS = -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/common  -I$(top_srcdir)/include/mtcr_ul
+USER_DIR = $(top_srcdir)
+AM_CPPFLAGS = -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/common  -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)
 
 AM_CFLAGS = -W -Wall  -g -MP -MD -Wswitch-enum $(COMPILER_FPIC) -DMTCR_EXPORT
 lib_LTLIBRARIES = libdev_mgt.a
@@ -50,9 +50,9 @@ dev_mgt_pylib_DATA = c_dev_mgt.so dev_mgt.py
 dist_dev_mgt_pylib_DATA = dev_mgt.py
 
 c_dev_mgt.so: libdev_mgt.a
-	$(CC) -g -Wall -pthread -shared ${CFLAGS} tools_dev_types.o -o c_dev_mgt.so \
-	-L$(USER_DIR)/reg_access -lreg_access \
-	-L$(USER_DIR)/tools_layouts -ltools_layouts \
-	-L$(USER_DIR)/${MTCR_CONF_DIR} -lmtcr_ul
+	$(CC) -g -Wall -pthread -shared ${CFLAGS} libdev_mgt.a -o c_dev_mgt.so \
+	-L$(top_builddir)/reg_access -lreg_access \
+	-L$(top_builddir)/tools_layouts -ltools_layouts \
+	-L$(top_builddir)/${MTCR_CONF_DIR} -lmtcr_ul
 
 CLEANFILES = c_dev_mgt.so
diff --git a/ext_libs/iniParser/Makefile.am b/ext_libs/iniParser/Makefile.am
index c4df7aa..1371b0f 100755
--- a/ext_libs/iniParser/Makefile.am
+++ b/ext_libs/iniParser/Makefile.am
@@ -32,7 +32,7 @@
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
 
-USER_DIR = ../..
+USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONFIG_DIR}
 INCLUDES = -I. -I$(USER_DIR) -I$(MTCR_DIR) -I$(USER_DIR)/common
 
diff --git a/flint/Makefile.am b/flint/Makefile.am
index d86f2c9..5c920ae 100644
--- a/flint/Makefile.am
+++ b/flint/Makefile.am
@@ -53,18 +53,18 @@ bin_PROGRAMS = mstflint
 mstflint_SOURCES = flint.cpp flint.h subcommands.cpp subcommands.h subcommands_linkx.cpp \
 					flint_params.cpp flint_params.h cmd_line_parser.cpp err_msgs.h
 
-mstflint_LDADD =  ../mlxfwops/lib/libmlxfwops.a  \
-                  ../cmdparser/libcmdparser.a    \
-		  ../mflash/libmflash.a          \
-		  ../tools_res_mgmt/libtools_res_mgmt.a \
-		  $(CMDIF_DIR)/libcmdif.a        \
-		  ../reg_access/libreg_access.a  \
-		  ../dev_mgt/libdev_mgt.a        \
-		  ../${MTCR_CONF_DIR}/libmtcr_ul.a        \
-		  ../tools_layouts/libtools_layouts.a \
-		  ../fw_comps_mgr/libfw_comps_mgr.a \
-		  ../mft_utils/libmftutils.a\
-		  ../mft_utils/hsmclient/libhsmclient.a\
+mstflint_LDADD =  $(top_builddir)/mlxfwops/lib/libmlxfwops.a  \
+                  $(top_builddir)/cmdparser/libcmdparser.a    \
+		  $(top_builddir)/mflash/libmflash.a          \
+		  $(top_builddir)/tools_res_mgmt/libtools_res_mgmt.a \
+		  $(top_builddir)/cmdif/libcmdif.a        \
+		  $(top_builddir)/reg_access/libreg_access.a  \
+		  $(top_builddir)/dev_mgt/libdev_mgt.a        \
+		  $(top_builddir)/${MTCR_CONF_DIR}/libmtcr_ul.a        \
+		  $(top_builddir)/tools_layouts/libtools_layouts.a \
+		  $(top_builddir)/fw_comps_mgr/libfw_comps_mgr.a \
+		  $(top_builddir)/mft_utils/libmftutils.a\
+		  $(top_builddir)/mft_utils/hsmclient/libhsmclient.a\
 		  ${LDL}
 
 
@@ -75,23 +75,23 @@ mstflint_CXXFLAGS += -DNO_ZLIB
 endif
 
 if ENABLE_CS
-mstflint_LDADD += $(top_srcdir)/tools_crypto/libtools_crypto.a -lcrypto
+mstflint_LDADD += $(top_builddir)/tools_crypto/libtools_crypto.a -lcrypto
 else
 endif
 
 if ENABLE_INBAND
-mstflint_LDADD += $(MAD_IFC_DIR)/libmad_ifc.a
+mstflint_LDADD += $(top_builddir)/mad_ifc/libmad_ifc.a
 else
 endif
 
 if ENABLE_OPENSSL
-mstflint_LDADD += $(top_srcdir)/mlxsign_lib/libmlxsign.a -lcrypto -lssl
+mstflint_LDADD += $(top_builddir)/mlxsign_lib/libmlxsign.a -lcrypto -lssl
 else
 mstflint_CXXFLAGS += -DNO_OPEN_SSL
 endif
 
 if ENABLE_FWMGR
-mstflint_LDADD += $(top_srcdir)/libmfa/libmfa.a $(top_srcdir)/ext_libs/minixz/libminixz.a $(top_srcdir)/mlxarchive/libmstarchive.a $(top_srcdir)/tools_layouts/libmlxarchive_layouts.a $(top_srcdir)/xz_utils/libxz_utils.a -llzma -lm
+mstflint_LDADD += $(top_builddir)/libmfa/libmfa.a $(top_builddir)/ext_libs/minixz/libminixz.a $(top_builddir)/mlxarchive/libmstarchive.a $(top_builddir)/tools_layouts/libmlxarchive_layouts.a $(top_builddir)/xz_utils/libxz_utils.a -llzma -lm
 else
 mstflint_CXXFLAGS += -DNO_MSTARCHIVE
 endif
diff --git a/fw_comps_mgr/Makefile.am b/fw_comps_mgr/Makefile.am
index 018e742..1c4b873 100755
--- a/fw_comps_mgr/Makefile.am
+++ b/fw_comps_mgr/Makefile.am
@@ -62,9 +62,9 @@ else
 AM_CXXFLAGS += "-DNO_INBAND"
 endif
 
-libfw_comps_mgr_a_LIBADD = ../reg_access/libreg_access.a $(MTCR_DIR)/libmtcr_ul.a $(LAYOUTS_LIB) $(TOOLS_RES_MGMT_DIR)/libtools_res_mgmt.a \
-                            $(MFT_UTILS_DIR)/libmftutils.a
+libfw_comps_mgr_a_LIBADD = $(top_builddir)/reg_access/libreg_access.a $(top_builddir)/$(MTCR_CONF_DIR)/libmtcr_ul.a $(LAYOUTS_LIB) $(top_builddir)/tools_res_mgmt/libtools_res_mgmt.a \
+                            $(top_builddir)/mft_utils/libmftutils.a
 
 if ENABLE_INBAND
-libfw_comps_mgr_a_LIBADD +=  $(top_srcdir)/mad_ifc/libmad_ifc.a
+libfw_comps_mgr_a_LIBADD +=  $(top_builddir)/mad_ifc/libmad_ifc.a
 endif
diff --git a/mad_ifc/Makefile.am b/mad_ifc/Makefile.am
index 40166c5..ace6b25 100644
--- a/mad_ifc/Makefile.am
+++ b/mad_ifc/Makefile.am
@@ -31,11 +31,11 @@
 #--
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONF_DIR}
 MTCR_INC_DIR = $(top_srcdir)/include/mtcr_ul
 
-INCLUDES = -I. -I.. -I$(MTCR_INC_DIR) -I../${MTCR_CONF_DIR} -I$(USER_DIR)/tools_ayouts/ 
+INCLUDES = -I. -I$(USER_DIR) -I$(MTCR_INC_DIR) -I$(USER_DIR)/${MTCR_CONF_DIR} -I$(USER_DIR)/tools_layouts/
 
 AM_CFLAGS = -W -Wall  -g -MP -MD $(COMPILER_FPIC)
 
@@ -43,6 +43,6 @@ AM_CFLAGS = -W -Wall  -g -MP -MD $(COMPILER_FPIC)
 noinst_LIBRARIES = libmad_ifc.a
 
 libmad_ifc_a_SOURCES = mad_ifc.c mad_ifc.h
-libmad_ifc_a_DEPENDENCIES = $(USER_DIR)/tools_layouts/libtools_layouts.a
+libmad_ifc_a_DEPENDENCIES = $(top_builddir)/tools_layouts/libtools_layouts.a
 libmad_ifc_a_LIBADD = $(libmad_ifc_a_DEPENDENCIES)
 
diff --git a/mflash/Makefile.am b/mflash/Makefile.am
index d57abaf..d45d353 100644
--- a/mflash/Makefile.am
+++ b/mflash/Makefile.am
@@ -32,7 +32,7 @@
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
 AM_CPPFLAGS = -I. -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/tools_layouts -I$(top_srcdir)/reg_access \
-          -I$(top_srcdir)/cmdif -I$(top_srcdir)/tools_res_mgmt $(COMPILER_FPIC)
+          -I$(top_srcdir)/cmdif -I$(top_srcdir)/tools_res_mgmt $(COMPILER_FPIC) -I$(top_srcdir)
 
 AM_CFLAGS = -MD -pipe -Wall -W  -DMST_UL -g ${MFLASH_INBAND_FLAG} $(COMPILER_FPIC)
 
diff --git a/mft_utils/hsmclient/Makefile.am b/mft_utils/hsmclient/Makefile.am
index 911d356..75f1030 100644
--- a/mft_utils/hsmclient/Makefile.am
+++ b/mft_utils/hsmclient/Makefile.am
@@ -32,7 +32,7 @@
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
 USER_DIR = $(top_srcdir)
-AM_CPPFLAGS = -I$(USER_DIR)/common -Iinc -Iinc/rsa $(COMPILER_FPIC) -DUNIX -DOS_UNIX -DOS_LINUX
+AM_CPPFLAGS = -I$(USER_DIR)/common -I$(srcdir)/inc -I$(srcdir)/inc/rsa $(COMPILER_FPIC) -DUNIX -DOS_UNIX -DOS_LINUX
 
 AM_CFLAGS = -Wall -W -g -MP -MD -pipe  $(COMPILER_FPIC)
 
diff --git a/mlxconfig/Makefile.am b/mlxconfig/Makefile.am
index 29ad8a4..a92aeda 100755
--- a/mlxconfig/Makefile.am
+++ b/mlxconfig/Makefile.am
@@ -73,13 +73,15 @@ mlxcfg_db_items.h mlxcfg_db_items.cpp\
         mlxcfg_commander.h mlxcfg_commander.cpp mlxcfg_generic_commander.h mlxcfg_generic_commander.cpp\
         mlxcfg_param_lib.h mlxcfg_param_lib.cpp mlxcfg_4th_gen_commander.h mlxcfg_4th_gen_commander.cpp
 
-libmlxcfg_a_LIBADD = $(UTILS_LIB) $(MUPARSER_LIBS) $(SQLITE_LIBS)\
-				$(CMDIF_DIR)/libcmdif.a ../reg_access/libreg_access.a ../reg_access/libmlxconfig_4th_gen_regsiters.a $(LAYOUTS_LIB) $(MLXCONFIG_4TH_GEN_LAYOUTS_LIB) $(MTCR_DIR)/libmtcr_ul.a\
-				$(DEV_MGT_DIR)/libdev_mgt.a $(COMPS_MGR_DIR)/libfw_comps_mgr.a $(TOOLS_RES_MGMT_DIR)/libtools_res_mgmt.a $(MLNXOS_PPC_LIBS) $(LIBSTD_CPP) ${LDL}
+libmlxcfg_a_LIBADD = $(top_builddir)/mft_utils/libmftutils.a $(MUPARSER_LIBS) $(SQLITE_LIBS)\
+				$(top_builddir)/cmdif/libcmdif.a $(top_builddir)/reg_access/libreg_access.a $(top_builddir)/reg_access/libmlxconfig_4th_gen_regsiters.a \
+                $(top_builddir)/tools_layouts/libtools_layouts.a $(top_builddir)/tools_layouts/libmlxconfig_4th_gen_layouts.a $(top_builddir)/$(MTCR_CONF_DIR)/libmtcr_ul.a\
+				$(top_builddir)/dev_mgt/libdev_mgt.a $(top_builddir)/fw_comps_mgr/libfw_comps_mgr.a $(top_builddir)/tools_res_mgmt/libtools_res_mgmt.a $(MLNXOS_PPC_LIBS) $(LIBSTD_CPP) ${LDL}
 
-mstconfig_LDADD = libmlxcfg.a $(UTILS_LIB) $(MUPARSER_LIBS) $(SQLITE_LIBS)\
-				$(CMDIF_DIR)/libcmdif.a ../reg_access/libreg_access.a ../reg_access/libmlxconfig_4th_gen_regsiters.a $(LAYOUTS_LIB) $(MLXCONFIG_4TH_GEN_LAYOUTS_LIB) $(MTCR_DIR)/libmtcr_ul.a\
-				$(DEV_MGT_DIR)/libdev_mgt.a $(COMPS_MGR_DIR)/libfw_comps_mgr.a $(TOOLS_RES_MGMT_DIR)/libtools_res_mgmt.a $(MLNXOS_PPC_LIBS) $(LIBSTD_CPP) ${LDL}
+mstconfig_LDADD = libmlxcfg.a $(top_builddir)/mft_utils/libmftutils.a $(MUPARSER_LIBS) $(SQLITE_LIBS)\
+				$(top_builddir)/cmdif/libcmdif.a $(top_builddir)/reg_access/libreg_access.a $(top_builddir)/reg_access/libmlxconfig_4th_gen_regsiters.a \
+                $(top_builddir)/tools_layouts/libtools_layouts.a $(top_builddir)/tools_layouts/libmlxconfig_4th_gen_layouts.a $(top_builddir)/$(MTCR_CONF_DIR)/libmtcr_ul.a\
+				$(top_builddir)/dev_mgt/libdev_mgt.a $(top_builddir)/fw_comps_mgr/libfw_comps_mgr.a $(top_builddir)/tools_res_mgmt/libtools_res_mgmt.a $(MLNXOS_PPC_LIBS) $(LIBSTD_CPP) ${LDL}
 
 if DISABLE_XML2
 AM_CXXFLAGS += -DDISABLE_XML2
@@ -89,14 +91,14 @@ libmlxcfg_a_LIBADD += -lxml2
 endif
 
 if ENABLE_INBAND
-mstconfig_LDADD += $(top_srcdir)/mad_ifc/libmad_ifc.a
-libmlxcfg_a_LIBADD += $(top_srcdir)/mad_ifc/libmad_ifc.a
+mstconfig_LDADD += $(top_builddir)/mad_ifc/libmad_ifc.a
+libmlxcfg_a_LIBADD += $(top_builddir)/mad_ifc/libmad_ifc.a
 else
 endif
 
 if ENABLE_OPENSSL
-mstconfig_LDADD +=   $(MLXSIGN_LIB) -lcrypto -lssl
-libmlxcfg_a_LIBADD +=   $(MLXSIGN_LIB) -lcrypto -lssl
+mstconfig_LDADD +=   $(top_builddir)/mlxsign_lib/libmlxsign.a -lcrypto -lssl
+libmlxcfg_a_LIBADD +=   $(top_builddir)/mlxsign_lib/libmlxsign.a -lcrypto -lssl
 else
 AM_CXXFLAGS += -DNO_OPEN_SSL
 endif
diff --git a/mlxfwops/lib/Makefile.am b/mlxfwops/lib/Makefile.am
index 296540f..4c06b31 100644
--- a/mlxfwops/lib/Makefile.am
+++ b/mlxfwops/lib/Makefile.am
@@ -47,7 +47,7 @@ COMPONENTS_DIR = components
 AM_CPPFLAGS = -I$(srcdir) -I$(MTCR_INC_DIR) -I$(MFLASH_DIR) $(JSON_CFLAGS) -I$(MINIXZ_DIR) \
               -I$(COMMON_DIR) -I$(MFT_EXT_LIBS_INC_DIR) -I$(MFT_EXT_LIBS_INC_DIR)/zlib -I$(LAYOUTS_DIR) \
               -I$(top_srcdir)/common -I$(UTILS_LIB) -I$(UEFI_COMMON_DIR) -I$(DEV_MGT_DIR) -I$(MTCR_UL_DIR) \
-              -I$(COMPONENTS_DIR)
+              -I$(COMPONENTS_DIR) -I$(top_srcdir) -I$(srcdir)/components
 
 MLXFWOPS_VERSION = 1
 
diff --git a/mlxfwupdate/Makefile.am b/mlxfwupdate/Makefile.am
index 4f28e63..6333b02 100755
--- a/mlxfwupdate/Makefile.am
+++ b/mlxfwupdate/Makefile.am
@@ -31,7 +31,7 @@
 #--
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONFIG_DIR}
 
 bin_PROGRAMS = mstfwmanager
diff --git a/mlxsign_lib/Makefile.am b/mlxsign_lib/Makefile.am
index 72e2e58..de0502d 100644
--- a/mlxsign_lib/Makefile.am
+++ b/mlxsign_lib/Makefile.am
@@ -31,7 +31,7 @@
 #--
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 INCLUDES = -I$(USER_DIR) -I$(USER_DIR)/common
 AM_CXXFLAGS = -Wall -W  -g -MP -MD -pipe $(COMPILER_FPIC)
 AM_CXXFLAGS += -DTOOLS_CRYPTO_KEY='$(TOOLS_CRYPTO_KEY)' -DTOOLS_CRYPTO_IV='$(TOOLS_CRYPTO_IV)'
diff --git a/mtcr_py/Makefile.am b/mtcr_py/Makefile.am
index 934b803..466c531 100644
--- a/mtcr_py/Makefile.am
+++ b/mtcr_py/Makefile.am
@@ -34,13 +34,13 @@
 # Makefile.am -- Process this file with automake to produce Makefile.in
 MTCR_CONFIG_DIR = mtcr_ul
 mtcr_pylibdir = $(libdir)/mstflint/python_tools/
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONF_DIR}
 
 mtcr_pylib_DATA = cmtcr.so mtcr.py
 dist_mtcr_pylib_DATA = mtcr.py
 cmtcr.so:
-	$(CC) -g -Wall -pthread -shared ${CFLAGS} $(MTCR_DIR)/*.o -o cmtcr.so $(MFT_CORE_LIB_LINK)
+	$(CC) -g -Wall -pthread -shared ${CFLAGS} $(top_builddir)/$(MTCR_CONF_DIR)/*.o -o cmtcr.so $(MFT_CORE_LIB_LINK)
 
 CLEANFILES = cmtcr.so
 
diff --git a/mtcr_ul/Makefile.am b/mtcr_ul/Makefile.am
index 479bf48..5295d5c 100644
--- a/mtcr_ul/Makefile.am
+++ b/mtcr_ul/Makefile.am
@@ -31,7 +31,7 @@
 #--
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
-AM_CPPFLAGS = -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/dev_mgt
+AM_CPPFLAGS = -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/dev_mgt -I$(top_srcdir)
 
 pkglib_LIBRARIES = libmtcr_ul.a
 
diff --git a/mvpd/Makefile.am b/mvpd/Makefile.am
index fc2d4ad..ebb64e4 100644
--- a/mvpd/Makefile.am
+++ b/mvpd/Makefile.am
@@ -43,7 +43,7 @@ noinst_LTLIBRARIES = libmvpd.a
 
 libmvpd_a_SOURCES =  mvpd.c mvpd.h
 
-libmvpd_a_DEPENDENCIES = $(MTCR_DIR)/libmtcr_ul.a
+libmvpd_a_DEPENDENCIES = $(top_builddir)/$(MTCR_CONF_DIR)/libmtcr_ul.a
 libmvpd_a_LIBADD = $(libmvpd_a_DEPENDENCIES)
 
 RMVPD_SO = rmvpd.so
@@ -55,7 +55,7 @@ LDADD= ../${MTCR_CONF_DIR}/libmtcr_ul.a ${LDL}
 noinst_HEADERS = mvpd.h
 
 ${RMVPD_SO}: libmvpd.a
-	$(CC) -g -Wall -pthread -shared ${CFLAGS} *.o -o ${RMVPD_SO} \
-	 -L$(MTCR_DIR) -lmtcr_ul
+	$(CC) -g -Wall -pthread -shared ${CFLAGS} libmvpd.a -o ${RMVPD_SO} \
+	 -L$(top_builddir)/$(MTCR_CONF_DIR) -lmtcr_ul
 	
 CLEANFILES = ${RMVPD_SO}
diff --git a/reg_access/Makefile.am b/reg_access/Makefile.am
index 9f989e3..fb2830a 100644
--- a/reg_access/Makefile.am
+++ b/reg_access/Makefile.am
@@ -43,11 +43,11 @@ AM_CFLAGS = -W -Wall -g -MP -MD $(COMPILER_FPIC)
 lib_LTLIBRARIES = libreg_access.a libmlxconfig_4th_gen_regsiters.a
 
 libreg_access_a_SOURCES = reg_access.c reg_access.h reg_access_common.h reg_access_macros.h
-libreg_access_a_DEPENDENCIES = $(USER_DIR)/tools_layouts/libtools_layouts.a
+libreg_access_a_DEPENDENCIES = $(top_builddir)/tools_layouts/libtools_layouts.a
 libreg_access_a_LIBADD = $(libreg_access_a_DEPENDENCIES)
 
 libmlxconfig_4th_gen_regsiters_a_SOURCES = mlxconfig_4th_gen_registers.c mlxconfig_4th_gen_registers.h
-libmlxconfig_4th_gen_regsiters_a_DEPENDENCIES = $(USER_DIR)/tools_layouts/libmlxconfig_4th_gen_layouts.a
+libmlxconfig_4th_gen_regsiters_a_DEPENDENCIES = $(top_builddir)/tools_layouts/libmlxconfig_4th_gen_layouts.a
 libmlxconfig_4th_gen_regsiters_a_LIBADD = $(libmlxconfig_4th_gen_regsiters_la_DEPENDENCIES)
 RREG_ACCESS_SO = rreg_access.so
 reg_access_pylibdir = $(libdir)/mstflint/python_tools/
@@ -56,7 +56,7 @@ reg_access_pylib_DATA = ${RREG_ACCESS_SO} regaccess.py
 dist_reg_access_pylib_DATA = regaccess.py
 
 ${RREG_ACCESS_SO}: libreg_access.a
-	$(CC) -g -Wall -pthread -shared ${CFLAGS} *.o -o ${RREG_ACCESS_SO} \
-	-L$(USER_DIR)/tools_layouts -ltools_layouts -lmlxconfig_4th_gen_layouts -L$(MTCR_DIR) -lmtcr_ul
+	$(CC) -g -Wall -pthread -shared ${CFLAGS} libreg_access.a -o ${RREG_ACCESS_SO} \
+	-L$(top_builddir)/tools_layouts -ltools_layouts -lmlxconfig_4th_gen_layouts -L$(top_builddir)/$(MTCR_CONF_DIR) -lmtcr_ul
 
 CLEANFILES = ${RREG_ACCESS_SO}
diff --git a/resourcedump/Makefile.am b/resourcedump/Makefile.am
index 4add840..3207d65 100755
--- a/resourcedump/Makefile.am
+++ b/resourcedump/Makefile.am
@@ -47,24 +47,24 @@ USER_DIR = $(top_srcdir)
 bin_SCRIPTS=mstresourcedump
 
 PYTHON_WRAPPER=mstresourcedump
-PYTHON_WRAPPER_SCRIPT=$(USER_DIR)/common/python_wrapper
+PYTHON_WRAPPER_SCRIPT=$(top_builddir)/common/python_wrapper
 ${PYTHON_WRAPPER}: $(PYTHON_WRAPPER_SCRIPT)
 	cp $(PYTHON_WRAPPER_SCRIPT) $@
 pythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump
-dist_pythonlib_DATA = __init__.py mstresourcedump.py
+dist_pythonlib_DATA = $(srcdir)/__init__.py mstresourcedump.py
 validationpythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/validation
-dist_validationpythonlib_DATA = validation/*.py
+dist_validationpythonlib_DATA = $(srcdir)/validation/*.py
 utilspythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/utils
-dist_utilspythonlib_DATA = utils/*.py
+dist_utilspythonlib_DATA = $(srcdir)/utils/*.py
 segmentspythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/segments
-dist_segmentspythonlib_DATA = segments/*.py
+dist_segmentspythonlib_DATA = $(srcdir)/segments/*.py
 resourcepythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/resource_data
-dist_resourcepythonlib_DATA = resource_data/*.py
+dist_resourcepythonlib_DATA = $(srcdir)/resource_data/*.py
 filterspythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/filters
-dist_filterspythonlib_DATA = filters/*.py 
+dist_filterspythonlib_DATA = $(srcdir)/filters/*.py 
 fetcherspythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/fetchers
-dist_fetcherspythonlib_DATA = fetchers/*.py
+dist_fetcherspythonlib_DATA = $(srcdir)/fetchers/*.py
 commandspythonlibdir=$(libdir)/mstflint/python_tools/mstresourcedump/commands
-dist_commandspythonlib_DATA = commands/*.py
+dist_commandspythonlib_DATA = $(srcdir)/commands/*.py
 
 CLEANFILES=${PYTHON_WRAPPER}
diff --git a/resourceparse/Makefile.am b/resourceparse/Makefile.am
index c7c0a58..675f231 100755
--- a/resourceparse/Makefile.am
+++ b/resourceparse/Makefile.am
@@ -35,18 +35,18 @@ USER_DIR = $(top_srcdir)
 bin_SCRIPTS=mstresourceparse
 
 PYTHON_WRAPPER=mstresourceparse
-PYTHON_WRAPPER_SCRIPT=$(USER_DIR)/common/python_wrapper
+PYTHON_WRAPPER_SCRIPT=$(top_builddir)/common/python_wrapper
 ${PYTHON_WRAPPER}: $(PYTHON_WRAPPER_SCRIPT)
 	cp $(PYTHON_WRAPPER_SCRIPT) $@
 pythonlibdir=$(libdir)/mstflint/python_tools/mstresourceparse
-dist_pythonlib_DATA = __init__.py mstresourceparse.py
+dist_pythonlib_DATA = $(srcdir)/__init__.py mstresourceparse.py
 parserspythonlibdir=$(libdir)/mstflint/python_tools/mstresourceparse/parsers
-dist_parserspythonlib_DATA = parsers/*.py
+dist_parserspythonlib_DATA = $(srcdir)/parsers/*.py
 utilspythonlibdir=$(libdir)/mstflint/python_tools/mstresourceparse/utils
-dist_utilspythonlib_DATA = utils/*.py
+dist_utilspythonlib_DATA = $(srcdir)/utils/*.py
 segmentspythonlibdir=$(libdir)/mstflint/python_tools/mstresourceparse/segments
-dist_segmentspythonlib_DATA = segments/*.py
+dist_segmentspythonlib_DATA = $(srcdir)/segments/*.py
 resourcepythonlibdir=$(libdir)/mstflint/python_tools/mstresourceparse/resource_data
-dist_resourcepythonlib_DATA = resource_data/*.py
+dist_resourcepythonlib_DATA = $(srcdir)/resource_data/*.py
 
 CLEANFILES=${PYTHON_WRAPPER}
diff --git a/small_utils/Makefile.am b/small_utils/Makefile.am
index e713032..906b28d 100644
--- a/small_utils/Makefile.am
+++ b/small_utils/Makefile.am
@@ -32,10 +32,10 @@
 
 PYTHON_WRAPPER_SCRIPT = ../common/python_wrapper
 
-AM_CPPFLAGS= -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/mvpd $(COMPILER_FPIC)
-AM_CFLAGS= -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/mvpd $(COMPILER_FPIC)
+AM_CPPFLAGS= -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/mvpd $(COMPILER_FPIC) -I$(top_srcdir)
+AM_CFLAGS= -I$(top_srcdir)/include/mtcr_ul -I$(top_srcdir)/common -I$(top_srcdir)/mvpd $(COMPILER_FPIC) -I$(top_srcdir)
 
-LDADD= $(top_srcdir)/${MTCR_CONF_DIR}/libmtcr_ul.a ${LDL}
+LDADD= $(top_builddir)/${MTCR_CONF_DIR}/libmtcr_ul.a ${LDL}
 
 bin_PROGRAMS = mstmread \
 					mstmwrite \
@@ -51,20 +51,20 @@ mstmread_SOURCES = mread.c
 mstmwrite_SOURCES = mwrite.c
 
 mstvpd_SOURCES = vpd.c
-mstvpd_LDADD = $(top_srcdir)/mvpd/libmvpd.a \
-               $(top_srcdir)/${MTCR_CONF_DIR}/libmtcr_ul.a \
-               $(top_srcdir)/dev_mgt/libdev_mgt.a \
-               $(top_srcdir)/reg_access/libreg_access.a \
-               $(top_srcdir)/tools_layouts/libtools_layouts.a \
+mstvpd_LDADD = $(top_builddir)/mvpd/libmvpd.a \
+               $(top_builddir)/${MTCR_CONF_DIR}/libmtcr_ul.a \
+               $(top_builddir)/dev_mgt/libdev_mgt.a \
+               $(top_builddir)/reg_access/libreg_access.a \
+               $(top_builddir)/tools_layouts/libtools_layouts.a \
                ${LDL}
 
 mstcongestion_SOURCES = congestion.cpp
-mstcongestion_LDADD = $(top_srcdir)/cmdif/libcmdif.a \
-					  $(top_srcdir)/dev_mgt/libdev_mgt.a \
-                      $(top_srcdir)/reg_access/libreg_access.a \
-                      $(top_srcdir)/tools_layouts/libtools_layouts.a \
-                      $(top_srcdir)/cmdparser/libcmdparser.a \
-                      $(top_srcdir)/${MTCR_CONF_DIR}/libmtcr_ul.a ${LDL}
+mstcongestion_LDADD = $(top_builddir)/cmdif/libcmdif.a \
+					  $(top_builddir)/dev_mgt/libdev_mgt.a \
+                      $(top_builddir)/reg_access/libreg_access.a \
+                      $(top_builddir)/tools_layouts/libtools_layouts.a \
+                      $(top_builddir)/cmdparser/libcmdparser.a \
+                      $(top_builddir)/${MTCR_CONF_DIR}/libmtcr_ul.a ${LDL}
 
 mstmcra_SOURCES  = mcra.c
 
diff --git a/tools_res_mgmt/Makefile.am b/tools_res_mgmt/Makefile.am
index 6d7ba5d..b47b3de 100644
--- a/tools_res_mgmt/Makefile.am
+++ b/tools_res_mgmt/Makefile.am
@@ -34,7 +34,7 @@
 
 USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/include/mtcr_ul
-AM_CPPFLAGS = -I. -I../common -I$(MTCR_DIR) -I.. -I$(USER_DIR)/${MTCR_CONF_DIR}
+AM_CPPFLAGS = -I. -I$(USER_DIR)/common -I$(MTCR_DIR) -I$(USER_DIR) -I$(USER_DIR)/${MTCR_CONF_DIR}
 
 AM_CFLAGS = -W -Wall  -g -MP -MD $(COMPILER_FPIC)
 noinst_LIBRARIES = libtools_res_mgmt.a
diff --git a/xz_utils/Makefile.am b/xz_utils/Makefile.am
index 71ba504..0fa0da2 100755
--- a/xz_utils/Makefile.am
+++ b/xz_utils/Makefile.am
@@ -32,7 +32,7 @@
 
 # Makefile.am -- Process this file with automake to produce Makefile.in
 
-USER_DIR = ..
+USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONF_DIR}
 INCLUDES = -I. -I$(USER_DIR) -I$(MTCR_DIR) -I$(USER_DIR)/common
 AM_CFLAGS = -W  -g -MP -MD ${COMPILER_FPIC} 
-- 
2.34.1

