From 18fc2e48e4ecaa099dbcf0cd68f73514486ccc5a Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 20 Jul 2023 17:41:30 -0600
Subject: Remove duplicate pylib_DATA files


diff --git a/dev_mgt/Makefile.am b/dev_mgt/Makefile.am
index 2df5d5a..cd02ad0 100644
--- a/dev_mgt/Makefile.am
+++ b/dev_mgt/Makefile.am
@@ -47,7 +47,6 @@ libdev_mgt_a_SOURCES = \
 dev_mgt_pylibdir = $(libdir)/mstflint/python_tools/
 
 dev_mgt_pylib_DATA = c_dev_mgt.so dev_mgt.py
-dist_dev_mgt_pylib_DATA = dev_mgt.py
 
 c_dev_mgt.so: libdev_mgt.a
 	$(CC) -g -Wall -pthread -shared ${CFLAGS} libdev_mgt.a -o c_dev_mgt.so \
diff --git a/mtcr_py/Makefile.am b/mtcr_py/Makefile.am
index 466c531..069bff4 100644
--- a/mtcr_py/Makefile.am
+++ b/mtcr_py/Makefile.am
@@ -38,7 +38,6 @@ USER_DIR = $(top_srcdir)
 MTCR_DIR = $(USER_DIR)/${MTCR_CONF_DIR}
 
 mtcr_pylib_DATA = cmtcr.so mtcr.py
-dist_mtcr_pylib_DATA = mtcr.py
 cmtcr.so:
 	$(CC) -g -Wall -pthread -shared ${CFLAGS} $(top_builddir)/$(MTCR_CONF_DIR)/*.o -o cmtcr.so $(MFT_CORE_LIB_LINK)
 
diff --git a/reg_access/Makefile.am b/reg_access/Makefile.am
index fb2830a..3315385 100644
--- a/reg_access/Makefile.am
+++ b/reg_access/Makefile.am
@@ -53,7 +53,6 @@ RREG_ACCESS_SO = rreg_access.so
 reg_access_pylibdir = $(libdir)/mstflint/python_tools/
 
 reg_access_pylib_DATA = ${RREG_ACCESS_SO} regaccess.py
-dist_reg_access_pylib_DATA = regaccess.py
 
 ${RREG_ACCESS_SO}: libreg_access.a
 	$(CC) -g -Wall -pthread -shared ${CFLAGS} libreg_access.a -o ${RREG_ACCESS_SO} \
-- 
2.34.1

