From f1fa6555d8d91736e74ed9a910700c72f9e9a465 Mon Sep 17 00:00:00 2001
From: JC Fillion-Robin <jcfr@nvidia.com>
Date: Tue, 10 Mar 2026 12:47:55 -0400
Subject: [PATCH] Fix Main CMakeLists.txt

Signed-off-by: JC Fillion-Robin <jcfr@nvidia.com>
---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4641d6a66..ba3b1b998 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,9 @@ project(${HOLOSCAN_PACKAGE_NAME}
     DESCRIPTION "Holoscan SDK"
     LANGUAGES C CXX
 )
-include(SetupCUDA) # CUDA Language enabled there after additional setup
+
+ # Enable CUDA language (needed for CMAKE_CUDA_COMPILER)
+ enable_language(CUDA)
 
 # If enabling google sanitizer
 if(HOLOSCAN_ENABLE_GOOGLE_SANITIZER)
@@ -329,7 +331,6 @@ list(APPEND HOLOSCAN_INSTALL_TARGETS
     holoinfer_utils
     infer_utils
     logger
-    matx
     data_loggers_async_console_logger
     data_loggers_basic_console_logger
     op_async_ping_rx
-- 
2.50.1

