From f98086cbe07b0c10a9f30f93af1adb29085da9df Mon Sep 17 00:00:00 2001
From: JC Fillion-Robin <jcfr@nvidia.com>
Date: Tue, 10 Mar 2026 12:47:55 -0400
Subject: [PATCH] Fix Main CMakeLists.txt

Signed-off-by: JC Fillion-Robin <jcfr@nvidia.com>
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6f6f2f05..cb91aa1e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,9 @@ project(${HOLOSCAN_PACKAGE_NAME}
     DESCRIPTION "Holoscan SDK"
     LANGUAGES C CXX
 )
-include(SetupCUDA) # CUDA Language enabled there after additional setup
+
+ # Enable CUDA language (needed for CMAKE_CUDA_COMPILER)
+ enable_language(CUDA)
 
 # If enabling google sanitizer
 if(HOLOSCAN_ENABLE_GOOGLE_SANITIZER)
@@ -323,8 +325,6 @@ list(APPEND HOLOSCAN_INSTALL_TARGETS
     holoinfer_utils
     infer_utils
     logger
-    matx
-    holoscan_matx
     data_loggers_async_console_logger
     data_loggers_basic_console_logger
     op_async_ping_rx
-- 
2.50.1 (Apple Git-155)

