From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Mon, 12 Jun 2023 15:38:29 -0600
Subject: Move required 3rdparty headers to holoscan-core


diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba6812792..1ef2fab41 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -226,7 +226,7 @@ install(DIRECTORY include/holoscan/
 # Copy bundled tl-expected headers
 install(DIRECTORY ${tl-expected_SOURCE_DIR}/include/tl/
     DESTINATION include/3rdparty/tl
-    COMPONENT holoscan-dependencies
+    COMPONENT holoscan-core
 )
 
 # Copy bundled dlpack headers
diff --git a/cmake/deps/fmt_rapids.cmake b/cmake/deps/fmt_rapids.cmake
index f102de453..6ce1feb29 100644
--- a/cmake/deps/fmt_rapids.cmake
+++ b/cmake/deps/fmt_rapids.cmake
@@ -34,6 +34,6 @@ if(fmt_ADDED)
     # Install the headers needed for development with the SDK
     install(DIRECTORY ${fmt_SOURCE_DIR}/include/fmt
         DESTINATION "include"
-        COMPONENT "holoscan-dependencies"
+        COMPONENT "holoscan-core"
         )
 endif()
-- 
2.34.1

