From ea9cc2446f50cc40483642d59157739a264b9e53 Mon Sep 17 00:00:00 2001
From: Julien JOMIER <jjomier@nvidia.com>
Date: Fri, 30 Jan 2026 18:00:00 -0800
Subject: [PATCH] Fix TensorRT include interface

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
---
 cmake/modules/FindTensorRT.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/FindTensorRT.cmake b/cmake/modules/FindTensorRT.cmake
index 4fbc500..84f4c1d 100644
--- a/cmake/modules/FindTensorRT.cmake
+++ b/cmake/modules/FindTensorRT.cmake
@@ -60,7 +60,7 @@ macro(find_trt_library libname)
     add_library(TensorRT::${libname} SHARED IMPORTED GLOBAL)
     set_target_properties(TensorRT::${libname} PROPERTIES
         IMPORTED_LOCATION "${TensorRT_${libname}_LIBRARY}"
-        INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${TensorRT_INCLUDE_DIR}"
+        INTERFACE_INCLUDE_DIRECTORIES "${TensorRT_INCLUDE_DIR}"
     )
   endif()
 endmacro()
