From c9d39992a1e497f709f263446d4634acd82c9f08 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Tue, 13 Jun 2023 12:10:48 -0600
Subject: Disable narrowing conversion warning as error


diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4ad91602..a81bf7ce 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -18,6 +18,7 @@
 # ##############################################################################
 # TODO (grelee): I removed -Wold-style-cast due to it raising errors for ajantv2 code in .cache. After restructuring we can add it back in
 add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Werror$<SEMICOLON>-Wall$<SEMICOLON>-Wextra>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=narrowing>)
 
 function(add_holoscan_library target)
     add_library(${target} SHARED ${ARGN})
-- 
2.34.1

