From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Tue, 13 Jun 2023 12:12:01 -0600
Subject: Remove CONFIG from Protobuf find_package


diff --git a/cmake/deps/grpc.cmake b/cmake/deps/grpc.cmake
index 360f64c..54b435c 100644
--- a/cmake/deps/grpc.cmake
+++ b/cmake/deps/grpc.cmake
@@ -75,7 +75,7 @@ endfunction()
 # GRPC_CPP_EXECUTABLE - the path to the grpc_cpp_plugin executable
 
 set(protobuf_MODULE_COMPATIBLE TRUE)
-find_package(Protobuf CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
 message(STATUS "Using protobuf ${Protobuf_VERSION}")
 
 if(CMAKE_CROSSCOMPILING)
-- 
2.34.1

