From 6934cb4b53df131dd16b6957cdea7f9f75d3bedf Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Mon, 27 Mar 2023 20:19:18 -0600
Subject: Remove extra CUDA toolchain include

---
 cmake/modules/SetupCUDA.cmake | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/cmake/modules/SetupCUDA.cmake b/cmake/modules/SetupCUDA.cmake
index 904be8e..647b516 100644
--- a/cmake/modules/SetupCUDA.cmake
+++ b/cmake/modules/SetupCUDA.cmake
@@ -28,11 +28,6 @@ list(APPEND CMAKE_MESSAGE_CONTEXT "buildconfig")
 
 message(STATUS "Configuring CUDA Compiler")
 
-# If a toolchain file was provided, include it now to read any CMAKE_CUDA variables it may define.
-if(DEFINED CMAKE_TOOLCHAIN_FILE)
-    include(${CMAKE_TOOLCHAIN_FILE})
-endif()
-
 # Setup CMAKE_CUDA_HOST_COMPILER
 if(NOT DEFINED CMAKE_CUDA_HOST_COMPILER)
     message(STATUS "Setting CUDA host compiler to use the same CXX compiler defined by CMAKE_CXX_COMPILER(${CMAKE_CXX_COMPILER})")
-- 
2.34.1

