From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Fri, 7 Jul 2023 10:28:46 -0600
Subject: Fix RPATH for holoinfer


diff --git a/modules/holoinfer/src/CMakeLists.txt b/modules/holoinfer/src/CMakeLists.txt
index 4dd564e..c05ed4d 100644
--- a/modules/holoinfer/src/CMakeLists.txt
+++ b/modules/holoinfer/src/CMakeLists.txt
@@ -31,7 +31,7 @@ if(HOLOSCAN_BUILD_LIBTORCH)
         EXPORT_NAME infer::torch
         SOVERSION ${PROJECT_VERSION_MAJOR}
         VERSION ${PROJECT_VERSION}
-        INSTALL_RPATH_USE_LINK_PATH true # find libtorch and torchvision in install tree
+        INSTALL_RPATH_USE_LINK_PATH FALSE
     )
     target_link_libraries(holoinfer_torch
         PRIVATE
@@ -62,7 +62,7 @@ if(HOLOSCAN_BUILD_ORT)
         EXPORT_NAME infer::onnx_runtime
         SOVERSION ${PROJECT_VERSION_MAJOR}
         VERSION ${PROJECT_VERSION}
-        INSTALL_RPATH_USE_LINK_PATH true # find onnxruntime in install tree
+        INSTALL_RPATH_USE_LINK_PATH FALSE
     )
     target_link_libraries(holoinfer_ort
         PRIVATE
-- 
2.34.1

