From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 24 Aug 2023 19:51:45 -0600
Subject: Fix ONNXRuntime include paths


diff --git a/cmake/modules/FindONNXRuntime.cmake b/cmake/modules/FindONNXRuntime.cmake
index 9f8c10a..e9b41fa 100644
--- a/cmake/modules/FindONNXRuntime.cmake
+++ b/cmake/modules/FindONNXRuntime.cmake
@@ -29,10 +29,9 @@
 
 # Find headers
 find_path(ONNXRuntime_INCLUDE_DIR
-  NAMES onnxruntime/onnxruntime_c_api.h
+  NAMES onnxruntime_c_api.h
 )
 list(APPEND ONNXRuntime_INCLUDE_DIRS "${ONNXRuntime_INCLUDE_DIR}")
-list(APPEND ONNXRuntime_INCLUDE_DIRS "${ONNXRuntime_INCLUDE_DIR}/onnxruntime")
 mark_as_advanced(ONNXRuntime_INCLUDE_DIR)
 mark_as_advanced(ONNXRuntime_INCLUDE_DIRS)
 
-- 
2.34.1

