From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Tue, 2 Apr 2024 19:26:06 -0600
Subject: Remove UCX header workaround


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1171d3e..f0d6f9d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -397,17 +397,6 @@ foreach(gxf_component IN LISTS _GXF_components)
   endif()
 endforeach()
 
-if(TARGET GXF::ucx)
-    # GXF UCX classes publicly depend on UCX headers.
-    # Workaround to include those headers without explicitly providing UCX targets.
-    # http://cdash.nvidia.com/viewBuildError.php?buildid=4461
-    set_property(
-        TARGET GXF::ucx
-        APPEND PROPERTY
-            INTERFACE_INCLUDE_DIRECTORIES "${PACKAGE_PREFIX_DIR}/include/3rdparty/ucx"
-    )
-endif()
-
 if(NOT TARGET GXF::gxe)
   add_executable(GXF::gxe IMPORTED)
   set_target_properties(GXF::gxe PROPERTIES
-- 
2.34.1

