From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Mon, 23 Sep 2024 13:40:53 -0600
Subject: Skip multiai_ultrasound data download


diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 29f7bb7..06898ae 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -36,20 +36,22 @@ if(HOLOSCAN_DOWNLOAD_DATASETS)
   add_dependencies(racerx_data racerx_download)
 endif()
 
-# Multiai ultrasound sample data
-add_library(multiai_ultrasound_data INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound/video/icardio_input1.avi")
-# Setup the installation rule
-install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound DESTINATION data COMPONENT holoscan-data)
-
-if(HOLOSCAN_DOWNLOAD_DATASETS)
-  # Download the multiai sample data
-  holoscan_download_data(multiai_ultrasound
-    URL https://api.ngc.nvidia.com/v2/resources/nvidia/clara-holoscan/holoscan_multi_ai_ultrasound_sample_data/versions/20221201/zip
-    URL_MD5 8096b080563e6a75c8e50c3b271364cd
-    DOWNLOAD_DIR ${CMAKE_CURRENT_SOURCE_DIR}
-    BYPRODUCTS "${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound/video/icardio_input1.avi"
-  )
-
-  # Add dependency to force the download
-  add_dependencies(multiai_ultrasound_data multiai_ultrasound_download)
+if(HOLOSCAN_BUILD_TESTS)
+  # Multiai ultrasound sample data
+  add_library(multiai_ultrasound_data INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound/video/icardio_input1.avi")
+  # Setup the installation rule
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound DESTINATION data COMPONENT holoscan-data)
+
+  if(HOLOSCAN_DOWNLOAD_DATASETS)
+    # Download the multiai sample data
+    holoscan_download_data(multiai_ultrasound
+      URL https://api.ngc.nvidia.com/v2/resources/nvidia/clara-holoscan/holoscan_multi_ai_ultrasound_sample_data/versions/20221201/zip
+      URL_MD5 8096b080563e6a75c8e50c3b271364cd
+      DOWNLOAD_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+      BYPRODUCTS "${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound/video/icardio_input1.avi"
+    )
+
+    # Add dependency to force the download
+    add_dependencies(multiai_ultrasound_data multiai_ultrasound_download)
+  endif()
 endif()
-- 
2.34.1

