From 9e8f1a0baadc9950126d99d0a8124d375809cd36 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Fri, 27 May 2022 10:36:55 -0600
Subject: [PATCH] WAR for cudalowlatencydemo build

---
 .../demoapps/NVIDIA/cudalowlatencydemo/CMakeLists.txt           | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ajaapps/crossplatform/demoapps/NVIDIA/cudalowlatencydemo/CMakeLists.txt b/ajaapps/crossplatform/demoapps/NVIDIA/cudalowlatencydemo/CMakeLists.txt
index 5de6713..cbc4740 100644
--- a/ajaapps/crossplatform/demoapps/NVIDIA/cudalowlatencydemo/CMakeLists.txt
+++ b/ajaapps/crossplatform/demoapps/NVIDIA/cudalowlatencydemo/CMakeLists.txt
@@ -5,6 +5,8 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
 	return()
 endif()
 
+find_package(Threads REQUIRED)
+
 find_package(CUDA 11.1)
 if (NOT CUDA_FOUND)
 	message("skipping cudalowlatencydemo (CUDA not found)")
-- 
2.17.1

