From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 29 Nov 2023 16:20:50 -0700
Subject: Disable dvplowlatencydemo for OE build


diff --git a/demos/NVIDIA/CMakeLists.txt b/demos/NVIDIA/CMakeLists.txt
index d4763867..b81af79f 100644
--- a/demos/NVIDIA/CMakeLists.txt
+++ b/demos/NVIDIA/CMakeLists.txt
@@ -1,2 +1 @@
 add_subdirectory(cudalowlatencydemo)
-add_subdirectory(dvplowlatencydemo)
-- 
2.34.1

