From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Mon, 22 Jul 2024 12:06:05 -0600
Subject: Use Holoscan boot images


diff --git a/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc b/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc
index aa72a17f..c67fce1c 100644
--- a/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc
+++ b/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc
@@ -236,9 +236,9 @@ FvNameGuid         = $(PLATFORM_GUID)
   # Logo Files
   #
   FILE FREEFORM = gNVIDIAPlatformLogoGuid {
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray480.bmp
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray720.bmp
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray1080.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-480.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-720.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-1080.bmp
   }
 !endif
 
-- 
2.34.1

