diff -ur edk2-tegra.a/edk2-nvidia/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc edk2-tegra.b/edk2-nvidia/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc
--- edk2-tegra.a/edk2-nvidia/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc	2023-03-30 09:12:05.255688674 -0600
+++ edk2-tegra.b/edk2-nvidia/Platform/NVIDIA/NVIDIA.fvmain.fdf.inc	2023-03-30 09:13:13.232046439 -0600
@@ -263,9 +263,9 @@
   # Logo Files
   #
   FILE FREEFORM = gNVIDIAPlatformLogoGuid {
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray480.bmp
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray720.bmp
-    SECTION RAW = Silicon/NVIDIA/Assets/nvidiagray1080.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-480.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-720.bmp
+    SECTION RAW = Silicon/NVIDIA/Assets/holoscan-1080.bmp
   }
 
 
