From 9a6c4b70cc1e8a7b071addd72c8f2403496e9248 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Fri, 7 Oct 2022 09:15:07 -0600
Subject: [PATCH] Makefile changes for MGX build

---
 Makefile | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index d13fd51..c96040d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 obj-m += nv_peer_mem.o
 
-PHONY += all clean install uninstall gen_nv_symvers
+PHONY += all clean install uninstall
 .PHONY: $(PHONY)
 
 KVER := $(shell uname -r)
@@ -27,7 +27,7 @@ MOD_NAME := nv_peer_mem
 MOD_VERSION = $(shell awk '/^Version:/ {print $$2}' nvidia_peer_memory.spec)
 DKMS_SRC_DIR = /usr/src/$(MOD_NAME)-$(MOD_VERSION)
 SOURCE_FILES := Makefile compat_nv-p2p.h nv_peer_mem.c \
-  create_nv.symvers.sh dkms.conf
+  dkms.conf
 
 # GCC earlier than 4.6.0 will build modules which require 'mcount',
 # and this symbol will not be available in the kernel if the kernel was
@@ -66,7 +66,7 @@ else
 NV_P2P_H=$(shell /bin/ls -1 /usr/src/nvidia-*/nvidia/nv-p2p.h 2>/dev/null | tail -1)
 endif
 
-all: gen_nv_symvers
+all:
 ifneq ($(shell test -e "$(NV_P2P_H)" && echo "true" || echo "" ),)
 	$(info Found $(NV_P2P_H))
 	/bin/cp -f $(NV_P2P_H) $(PWD)/nv-p2p.h
@@ -75,17 +75,17 @@ else
 	/bin/cp -f $(PWD)/compat_nv-p2p.h $(PWD)/nv-p2p.h
 endif
 	echo -n "" > my.symvers
-ifneq ($(shell test -d $(OFA_KERNEL) && echo "true" || echo "" ),)
-	# get OFED symbols when building with MLNX_OFED
-	/bin/cp -f $(OFA_KERNEL)/Module.symvers my.symvers
-endif
-	cat nv.symvers >> my.symvers
+	/bin/cp -f $(OFA_SYMVERS) my.symvers
+	cat ${NVIDIA_SYMVERS} >> my.symvers
 	make -C $(KDIR) $(MAKE_PARAMS) M=$(PWD) KBUILD_EXTRA_SYMBOLS="$(PWD)/my.symvers" modules
 
 clean:
 	make -C $(KDIR)  M=$(PWD) clean
 	/bin/rm -f nv.symvers my.symvers nv-p2p.h
 
+modules_install:
+	make -C $(KDIR)  M=$(PWD) modules_install
+
 install:
 	mkdir -p $(DESTDIR)/$(MODULE_DESTDIR);
 	/bin/cp -f $(PWD)/nv_peer_mem.ko $(DESTDIR)/$(MODULE_DESTDIR);
@@ -106,6 +106,3 @@ install-utils:
 uninstall:
 	/bin/rm -f $(DESTDIR)/$(MODULE_DESTDIR)/nv_peer_mem.ko
 	if [ ! -n "$(DESTDIR)" ]; then $(DEPMOD) -r -ae $(KVER);fi;
-
-gen_nv_symvers:
-	$(PWD)/create_nv.symvers.sh $(KVER)
-- 
2.17.1

