From dfc2e61a0ff83a6abf3e6db69903a1ff2d79d6ea Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 28 Apr 2025 10:55:04 +0100
Subject: [PATCH] Fixups for cross building in OE

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt | 46 ----------------------------------------------
 1 file changed, 46 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23be699..af02bf1 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -163,39 +163,6 @@ set(CUTLASS_ENABLE_SELF_CONTAINED_INCLUDES_CHECK ON CACHE BOOL "Enable CUTLASS c
 
 ################################################################################
 
-set(CUTLASS_NVCC_ARCHS_SUPPORTED "")
-if (CUDA_VERSION VERSION_GREATER_EQUAL 11.4)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 70 72 75 80 86 87)
-endif()
-if (CUDA_VERSION VERSION_GREATER_EQUAL 11.8)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 89 90)
-endif()
-if (CUDA_VERSION VERSION_GREATER_EQUAL 12.0)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 90a)
-endif()
-
-if (CUDA_VERSION VERSION_GREATER_EQUAL 12.8)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 100 100a 120 120a 121 121a)
-  if (CUDA_VERSION VERSION_LESS 13.0)
-    list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 101 101a)
-  else()
-    list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 110 110a)
-  endif()
-endif()
-
-if (CUDA_VERSION VERSION_GREATER_EQUAL 12.9)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 100f 120f 121f 103a 103f)
-  if (CUDA_VERSION VERSION_LESS 13.0)
-    list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 101f)
-  else()
-    list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 110f)
-  endif()
-endif()
-
-if (CUDA_VERSION VERSION_GREATER_EQUAL 13.0)
-  list(APPEND CUTLASS_NVCC_ARCHS_SUPPORTED 110 110a)
-endif()
-
 set(CUTLASS_NVCC_ARCHS ${CUTLASS_NVCC_ARCHS_SUPPORTED} CACHE STRING "The SM architectures requested.")
 set(CUTLASS_NVCC_ARCHS_ENABLED ${CUTLASS_NVCC_ARCHS} CACHE STRING "The SM architectures to build code for.")
 
@@ -800,19 +767,6 @@ if (DOXYGEN_FOUND)
     )
 endif()
 
-if(NOT WIN32)
-  # Add common library search paths so executables and libraries can load and run
-  # without LD_LIBRARY_PATH being set.
-  link_libraries(
-    "-Wl,-rpath,'$ORIGIN'"
-    "-Wl,-rpath,'$ORIGIN/../lib64'"
-    "-Wl,-rpath,'$ORIGIN/../lib'"
-    "-Wl,-rpath,'${CUDA_TOOLKIT_ROOT_DIR}/lib64'"
-    "-Wl,-rpath,'${CUDA_TOOLKIT_ROOT_DIR}/lib'"
-    ${CMAKE_DL_LIBS}
-    )
-endif()
-
 ################################################################################
 
 include(CTest)
-- 
2.34.1

