From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Mar 2023 19:09:23 -0700
Subject: Add install rules


diff --git a/applications/colonoscopy_segmentation/CMakeLists.txt b/applications/colonoscopy_segmentation/CMakeLists.txt
index 4ccd1b5..6ae0453 100644
--- a/applications/colonoscopy_segmentation/CMakeLists.txt
+++ b/applications/colonoscopy_segmentation/CMakeLists.txt
@@ -33,6 +33,14 @@ if(HOLOHUB_DOWNLOAD_DATASETS)
     )
 endif()
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/colonoscopy_segmentation.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/colonoscopy_segmentation.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # To get the environment path
diff --git a/applications/endoscopy_out_of_body_detection/CMakeLists.txt b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
index 8ad93dd..e8e4fd0 100644
--- a/applications/endoscopy_out_of_body_detection/CMakeLists.txt
+++ b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
@@ -60,6 +60,15 @@ add_custom_target(endoscopy_out_of_body_detection_yaml
 )
 add_dependencies(endoscopy_out_of_body_detection endoscopy_out_of_body_detection_yaml)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS endoscopy_out_of_body_detection
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/endoscopy_out_of_body_detection.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # Configure the yaml file to only play 10 frames
diff --git a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
index 4995322..958f220 100644
--- a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
+++ b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
@@ -52,6 +52,15 @@ add_custom_target(endoscopy_tool_tracking_deps
 )
 add_dependencies(endoscopy_tool_tracking endoscopy_tool_tracking_deps)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS endoscopy_tool_tracking
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/endoscopy_tool_tracking.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
 
diff --git a/applications/endoscopy_tool_tracking/python/CMakeLists.txt b/applications/endoscopy_tool_tracking/python/CMakeLists.txt
index 007f5a5..d3ce4af 100644
--- a/applications/endoscopy_tool_tracking/python/CMakeLists.txt
+++ b/applications/endoscopy_tool_tracking/python/CMakeLists.txt
@@ -13,6 +13,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/endoscopy_tool_tracking.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/endoscopy_tool_tracking.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # To get the environment path
diff --git a/applications/h264/h264_endoscopy_tool_tracking/CMakeLists.txt b/applications/h264/h264_endoscopy_tool_tracking/CMakeLists.txt
index df5f274..ca048d6 100644
--- a/applications/h264/h264_endoscopy_tool_tracking/CMakeLists.txt
+++ b/applications/h264/h264_endoscopy_tool_tracking/CMakeLists.txt
@@ -58,6 +58,15 @@ if(HOLOHUB_DOWNLOAD_DATASETS)
   add_dependencies(h264_endoscopy_tool_tracking endoscopy_data)
 endif()
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS h264_endoscopy_tool_tracking
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/h264_endoscopy_tool_tracking.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   add_test(NAME h264_endoscopy_tool_tracking_cpp_test
diff --git a/applications/h264/h264_video_decode/CMakeLists.txt b/applications/h264/h264_video_decode/CMakeLists.txt
index b097c1e..ee537b2 100644
--- a/applications/h264/h264_video_decode/CMakeLists.txt
+++ b/applications/h264/h264_video_decode/CMakeLists.txt
@@ -52,6 +52,15 @@ if(HOLOHUB_DOWNLOAD_DATASETS)
   add_dependencies(h264_video_decode endoscopy_data)
 endif()
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS h264_video_decode
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/h264_video_decode.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   add_test(NAME h264_video_decode_cpp_test
diff --git a/applications/high_speed_endoscopy/cpp/CMakeLists.txt b/applications/high_speed_endoscopy/cpp/CMakeLists.txt
index ea261e0..311a441 100644
--- a/applications/high_speed_endoscopy/cpp/CMakeLists.txt
+++ b/applications/high_speed_endoscopy/cpp/CMakeLists.txt
@@ -47,6 +47,15 @@ add_custom_target(high_speed_endoscopy_yaml
 )
 add_dependencies(high_speed_endoscopy high_speed_endoscopy_yaml)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS high_speed_endoscopy
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/high_speed_endoscopy.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # Warning: this test just check that the application starts if there is no camera
diff --git a/applications/high_speed_endoscopy/python/CMakeLists.txt b/applications/high_speed_endoscopy/python/CMakeLists.txt
index bc802e7..2fe1db8 100644
--- a/applications/high_speed_endoscopy/python/CMakeLists.txt
+++ b/applications/high_speed_endoscopy/python/CMakeLists.txt
@@ -13,6 +13,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/high_speed_endoscopy.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/high_speed_endoscopy.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # To get the environment path
diff --git a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
index a06560a..f0c93ea 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
@@ -49,6 +49,16 @@ add_custom_target(app_config_yaml
 )
 add_dependencies(multiai_endoscopy app_config_yaml)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS multiai_endoscopy
+  DESTINATION ${install_path}
+)
+install(FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/app_config.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
 
diff --git a/applications/multiai_endoscopy/python/CMakeLists.txt b/applications/multiai_endoscopy/python/CMakeLists.txt
index 38e39d0..d0fac8e 100644
--- a/applications/multiai_endoscopy/python/CMakeLists.txt
+++ b/applications/multiai_endoscopy/python/CMakeLists.txt
@@ -13,6 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/multi_ai.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/multi_ai.yaml
+  ${CMAKE_CURRENT_SOURCE_DIR}/endo_ref_data_labels.csv
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
 
diff --git a/applications/multiai_ultrasound/cpp/CMakeLists.txt b/applications/multiai_ultrasound/cpp/CMakeLists.txt
index d4c9196..8d845ee 100644
--- a/applications/multiai_ultrasound/cpp/CMakeLists.txt
+++ b/applications/multiai_ultrasound/cpp/CMakeLists.txt
@@ -55,6 +55,17 @@ add_custom_target(mgpu_multiai_ultrasound_yaml
 )
 add_dependencies(multiai_ultrasound mgpu_multiai_ultrasound_yaml)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS multiai_ultrasound
+  DESTINATION ${install_path}
+)
+install(FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound.yaml
+    ${CMAKE_CURRENT_SOURCE_DIR}/mgpu_multiai_ultrasound.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # Configure the yaml file to only play 10 frames
diff --git a/applications/multiai_ultrasound/python/CMakeLists.txt b/applications/multiai_ultrasound/python/CMakeLists.txt
index d18eb31..15a0a27 100644
--- a/applications/multiai_ultrasound/python/CMakeLists.txt
+++ b/applications/multiai_ultrasound/python/CMakeLists.txt
@@ -13,6 +13,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/multiai_ultrasound.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # To get the environment path
diff --git a/applications/object_detection_torch/CMakeLists.txt b/applications/object_detection_torch/CMakeLists.txt
index adfde26..4e5b81b 100644
--- a/applications/object_detection_torch/CMakeLists.txt
+++ b/applications/object_detection_torch/CMakeLists.txt
@@ -66,3 +66,12 @@ add_custom_target(object_detection_torch_yaml
     BYPRODUCTS "object_detection_torch.yaml" "postprocessing.yaml" "frcnn_resnet50_t.yaml" "labels.txt"
 )
 add_dependencies(object_detection_torch object_detection_torch_yaml)
+
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS object_detection_torch
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/object_detection_torch.yaml
+  DESTINATION ${install_path}
+)
diff --git a/applications/ultrasound_segmentation/cpp/CMakeLists.txt b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
index 0f1fff3..97e8a4d 100644
--- a/applications/ultrasound_segmentation/cpp/CMakeLists.txt
+++ b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
@@ -47,6 +47,15 @@ add_custom_target(ultrasound_segmentation_yaml
 )
 add_dependencies(ultrasound_segmentation ultrasound_segmentation_yaml)
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS ultrasound_segmentation
+  DESTINATION ${install_path}
+)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/ultrasound_segmentation.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
 
diff --git a/applications/ultrasound_segmentation/python/CMakeLists.txt b/applications/ultrasound_segmentation/python/CMakeLists.txt
index 14c60e2..fbe03e6 100644
--- a/applications/ultrasound_segmentation/python/CMakeLists.txt
+++ b/applications/ultrasound_segmentation/python/CMakeLists.txt
@@ -13,6 +13,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/ultrasound_segmentation.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/ultrasound_segmentation.yaml
+  DESTINATION ${install_path}
+)
+
 # Add testing
 if(BUILD_TESTING)
   # To get the environment path
diff --git a/applications/volume_rendering/CMakeLists.txt b/applications/volume_rendering/CMakeLists.txt
index df8d721..aeb189f 100644
--- a/applications/volume_rendering/CMakeLists.txt
+++ b/applications/volume_rendering/CMakeLists.txt
@@ -54,3 +54,9 @@ if(BUILD_TESTING)
   set_tests_properties(volume_rendering_test PROPERTIES
                        PASS_REGULAR_EXPRESSION "Application has finished running.")
 endif()
+
+# Install the application
+file(RELATIVE_PATH install_path ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+install(TARGETS volume_rendering
+  DESTINATION ${install_path}
+)
diff --git a/cmake/pybind11_add_holohub_module.cmake b/cmake/pybind11_add_holohub_module.cmake
index cb9bd6b..3f30510 100644
--- a/cmake/pybind11_add_holohub_module.cmake
+++ b/cmake/pybind11_add_holohub_module.cmake
@@ -80,4 +80,9 @@ function(pybind11_add_holohub_module)
         OUTPUT_NAME _${MODULE_NAME}
     )
 
-endfunction()
\ No newline at end of file
+    # Install the module
+    install(DIRECTORY ${HOLOHUB_PYTHON_MODULE_OUT_DIR}/${MODULE_NAME}
+        DESTINATION python/${CMAKE_INSTALL_LIBDIR}/holohub
+    )
+
+endfunction()
diff --git a/gxf_extensions/emergent_source/CMakeLists.txt b/gxf_extensions/emergent_source/CMakeLists.txt
index 4e68bb5..be77b47 100644
--- a/gxf_extensions/emergent_source/CMakeLists.txt
+++ b/gxf_extensions/emergent_source/CMakeLists.txt
@@ -50,3 +50,5 @@ add_library(gxf_emergent_source SHARED
 target_link_libraries(gxf_emergent_source
   PUBLIC gxf_emergent_source_lib
 )
+
+install(TARGETS gxf_emergent_source_lib gxf_emergent_source)
diff --git a/gxf_extensions/lstm_tensor_rt_inference/CMakeLists.txt b/gxf_extensions/lstm_tensor_rt_inference/CMakeLists.txt
index 8297091..d1cf650 100644
--- a/gxf_extensions/lstm_tensor_rt_inference/CMakeLists.txt
+++ b/gxf_extensions/lstm_tensor_rt_inference/CMakeLists.txt
@@ -43,3 +43,5 @@ add_library(gxf_lstm_tensor_rt_inference SHARED
 target_link_libraries(gxf_lstm_tensor_rt_inference
   PUBLIC gxf_lstm_tensor_rt_inference_lib
 )
+
+install(TARGETS gxf_lstm_tensor_rt_inference_lib gxf_lstm_tensor_rt_inference)
diff --git a/operators/emergent_source/CMakeLists.txt b/operators/emergent_source/CMakeLists.txt
index b64a454..a8db640 100644
--- a/operators/emergent_source/CMakeLists.txt
+++ b/operators/emergent_source/CMakeLists.txt
@@ -27,6 +27,8 @@ target_include_directories(emergent_source INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}
 
 target_link_libraries(emergent_source holoscan::core)
 
+install(TARGETS emergent_source)
+
 if(HOLOHUB_BUILD_PYTHON)
     add_subdirectory(python)
 endif()
diff --git a/operators/lstm_tensor_rt_inference/CMakeLists.txt b/operators/lstm_tensor_rt_inference/CMakeLists.txt
index 7bcbf41..995d929 100644
--- a/operators/lstm_tensor_rt_inference/CMakeLists.txt
+++ b/operators/lstm_tensor_rt_inference/CMakeLists.txt
@@ -26,6 +26,8 @@ add_library(lstm_tensor_rt_inference SHARED
 target_link_libraries(lstm_tensor_rt_inference holoscan::core)
 target_include_directories(lstm_tensor_rt_inference INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
+install(TARGETS lstm_tensor_rt_inference)
+
 if(HOLOHUB_BUILD_PYTHON)
     add_subdirectory(python)
 endif()
diff --git a/operators/tensor_to_video_buffer/CMakeLists.txt b/operators/tensor_to_video_buffer/CMakeLists.txt
index b73573d..964822c 100644
--- a/operators/tensor_to_video_buffer/CMakeLists.txt
+++ b/operators/tensor_to_video_buffer/CMakeLists.txt
@@ -30,3 +30,5 @@ target_link_libraries(
   tensor_to_video_buffer
   holoscan::core
 )
+
+install(TARGETS tensor_to_video_buffer)
diff --git a/operators/tool_tracking_postprocessor/CMakeLists.txt b/operators/tool_tracking_postprocessor/CMakeLists.txt
index 19e1767..8b471a1 100644
--- a/operators/tool_tracking_postprocessor/CMakeLists.txt
+++ b/operators/tool_tracking_postprocessor/CMakeLists.txt
@@ -30,6 +30,8 @@ set_target_properties(tool_tracking_postprocessor PROPERTIES CUDA_ARCHITECTURES
 target_link_libraries(tool_tracking_postprocessor holoscan::core)
 target_include_directories(tool_tracking_postprocessor INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
+install(TARGETS tool_tracking_postprocessor)
+
 if(HOLOHUB_BUILD_PYTHON)
     add_subdirectory(python)
 endif()
diff --git a/operators/video_decoder/CMakeLists.txt b/operators/video_decoder/CMakeLists.txt
index 0df2563..93b3c4c 100644
--- a/operators/video_decoder/CMakeLists.txt
+++ b/operators/video_decoder/CMakeLists.txt
@@ -31,3 +31,5 @@ add_library(holoscan::ops::video_decoder ALIAS video_decoder)
 target_include_directories(video_decoder INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(video_decoder holoscan::core)
+
+install(TARGETS video_decoder)
diff --git a/operators/video_encoder/CMakeLists.txt b/operators/video_encoder/CMakeLists.txt
index dd34904..a2ef42d 100644
--- a/operators/video_encoder/CMakeLists.txt
+++ b/operators/video_encoder/CMakeLists.txt
@@ -31,3 +31,5 @@ add_library(holoscan::ops::video_encoder ALIAS video_encoder)
 target_include_directories(video_encoder INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(video_encoder holoscan::core)
+
+install(TARGETS video_encoder)
diff --git a/operators/video_read_bitstream/CMakeLists.txt b/operators/video_read_bitstream/CMakeLists.txt
index 6ec8c1b..d45225e 100644
--- a/operators/video_read_bitstream/CMakeLists.txt
+++ b/operators/video_read_bitstream/CMakeLists.txt
@@ -27,3 +27,5 @@ add_library(holoscan::ops::video_read_bitstream ALIAS video_read_bitstream)
 target_include_directories(video_read_bitstream INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(video_read_bitstream holoscan::core)
+
+install(TARGETS video_read_bitstream)
diff --git a/operators/video_write_bitstream/CMakeLists.txt b/operators/video_write_bitstream/CMakeLists.txt
index ff4af21..5283c13 100644
--- a/operators/video_write_bitstream/CMakeLists.txt
+++ b/operators/video_write_bitstream/CMakeLists.txt
@@ -27,3 +27,5 @@ add_library(holoscan::ops::video_write_bitstream ALIAS video_write_bitstream)
 target_include_directories(video_write_bitstream INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(video_write_bitstream holoscan::core)
+
+install(TARGETS video_write_bitstream)
diff --git a/operators/visualizer_icardio/CMakeLists.txt b/operators/visualizer_icardio/CMakeLists.txt
index 65367ec..91e5730 100644
--- a/operators/visualizer_icardio/CMakeLists.txt
+++ b/operators/visualizer_icardio/CMakeLists.txt
@@ -32,6 +32,8 @@ target_include_directories(visualizer_icardio INTERFACE ${CMAKE_CURRENT_SOURCE_D
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MAJOR_VERSION=${holoscan_VERSION_MAJOR})
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MINOR_VERSION=${holoscan_VERSION_MINOR})
 
+install(TARGETS visualizer_icardio)
+
 if(HOLOHUB_BUILD_PYTHON)
     add_subdirectory(python)
 endif()
diff --git a/operators/volume_loader/CMakeLists.txt b/operators/volume_loader/CMakeLists.txt
index 597be6d..ee110f0 100644
--- a/operators/volume_loader/CMakeLists.txt
+++ b/operators/volume_loader/CMakeLists.txt
@@ -64,3 +64,5 @@ target_link_libraries(volume_loader
 target_include_directories(volume_loader INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_definitions(volume_loader PRIVATE HOLOSCAN_MAJOR_VERSION=${holoscan_VERSION_MAJOR})
 target_compile_definitions(volume_loader PRIVATE HOLOSCAN_MINOR_VERSION=${holoscan_VERSION_MINOR})
+
+install(TARGETS volume_loader)
diff --git a/operators/volume_renderer/CMakeLists.txt b/operators/volume_renderer/CMakeLists.txt
index f979e0f..faf6569 100644
--- a/operators/volume_renderer/CMakeLists.txt
+++ b/operators/volume_renderer/CMakeLists.txt
@@ -86,3 +86,5 @@ target_link_libraries(volume_renderer
 target_include_directories(volume_renderer INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_definitions(volume_renderer PRIVATE HOLOSCAN_MAJOR_VERSION=${holoscan_VERSION_MAJOR})
 target_compile_definitions(volume_renderer PRIVATE HOLOSCAN_MINOR_VERSION=${holoscan_VERSION_MINOR})
+
+install(TARGETS volume_renderer)
-- 
2.34.1

