From 436f6e1724f0e690fbae8cd68fe116a6af41fb4d Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Mar 2023 17:02:57 -0700
Subject: Fix undefined fmt errors


diff --git a/operators/tensor_to_video_buffer/CMakeLists.txt b/operators/tensor_to_video_buffer/CMakeLists.txt
index 964822c..9dfff8c 100644
--- a/operators/tensor_to_video_buffer/CMakeLists.txt
+++ b/operators/tensor_to_video_buffer/CMakeLists.txt
@@ -31,4 +31,6 @@ target_link_libraries(
   holoscan::core
 )
 
+target_compile_definitions(tensor_to_video_buffer PUBLIC FMT_HEADER_ONLY)
+
 install(TARGETS tensor_to_video_buffer)
diff --git a/operators/tool_tracking_postprocessor/CMakeLists.txt b/operators/tool_tracking_postprocessor/CMakeLists.txt
index 8b471a1..9467998 100644
--- a/operators/tool_tracking_postprocessor/CMakeLists.txt
+++ b/operators/tool_tracking_postprocessor/CMakeLists.txt
@@ -30,6 +30,8 @@ set_target_properties(tool_tracking_postprocessor PROPERTIES CUDA_ARCHITECTURES
 target_link_libraries(tool_tracking_postprocessor holoscan::core)
 target_include_directories(tool_tracking_postprocessor INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 
+target_compile_definitions(tool_tracking_postprocessor PUBLIC FMT_HEADER_ONLY)
+
 install(TARGETS tool_tracking_postprocessor)
 
 if(HOLOHUB_BUILD_PYTHON)
diff --git a/operators/visualizer_icardio/CMakeLists.txt b/operators/visualizer_icardio/CMakeLists.txt
index 91e5730..73c508a 100644
--- a/operators/visualizer_icardio/CMakeLists.txt
+++ b/operators/visualizer_icardio/CMakeLists.txt
@@ -32,6 +32,8 @@ target_include_directories(visualizer_icardio INTERFACE ${CMAKE_CURRENT_SOURCE_D
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MAJOR_VERSION=${holoscan_VERSION_MAJOR})
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MINOR_VERSION=${holoscan_VERSION_MINOR})
 
+target_compile_definitions(visualizer_icardio PUBLIC FMT_HEADER_ONLY)
+
 install(TARGETS visualizer_icardio)
 
 if(HOLOHUB_BUILD_PYTHON)
-- 
2.34.1

