From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Mar 2023 15:58:05 -0700
Subject: Fix ajantv2 dependencies


diff --git a/applications/endoscopy_out_of_body_detection/CMakeLists.txt b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
index ffaa4f5a..ca6fe315 100644
--- a/applications/endoscopy_out_of_body_detection/CMakeLists.txt
+++ b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
@@ -20,6 +20,8 @@ project(endoscopy_out_of_body_detection CXX)
 find_package(holoscan 1.0 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 # Download the associated dataset if needed
 if(HOLOHUB_DOWNLOAD_DATASETS)
   include(holoscan_download_data)
diff --git a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
index 2d16262f..49e5087f 100644
--- a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
+++ b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(endoscopy_tool_tracking CXX)
 find_package(holoscan 1.0 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(endoscopy_tool_tracking
   main.cpp
 )
diff --git a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
index 55b69339..b14c558d 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multiai_endoscopy CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(multiai_endoscopy
   multi_ai.cpp
 )
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
index debe8016..ce38d731 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
 enable_language(CUDA)
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
index 84c413fc..3dc21d15 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
 enable_language(CUDA)
diff --git a/applications/multiai_ultrasound/cpp/CMakeLists.txt b/applications/multiai_ultrasound/cpp/CMakeLists.txt
index 01e67ea3..2166fef0 100644
--- a/applications/multiai_ultrasound/cpp/CMakeLists.txt
+++ b/applications/multiai_ultrasound/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(multiai_ultrasound CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(multiai_ultrasound
     main.cpp
 )
diff --git a/applications/object_detection_torch/CMakeLists.txt b/applications/object_detection_torch/CMakeLists.txt
index a392485b..24529297 100644
--- a/applications/object_detection_torch/CMakeLists.txt
+++ b/applications/object_detection_torch/CMakeLists.txt
@@ -19,6 +19,8 @@ project(object_detection_torch CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(object_detection_torch
     main.cpp
 )
diff --git a/applications/ultrasound_segmentation/cpp/CMakeLists.txt b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
index 83d41ca4..ee28a979 100644
--- a/applications/ultrasound_segmentation/cpp/CMakeLists.txt
+++ b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(ultrasound_segmentation CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(ultrasound_segmentation
     main.cpp
 )
diff --git a/operators/aja_source/CMakeLists.txt b/operators/aja_source/CMakeLists.txt
index 1e5bda5a..9b7d9a03 100644
--- a/operators/aja_source/CMakeLists.txt
+++ b/operators/aja_source/CMakeLists.txt
@@ -18,28 +18,7 @@ project(aja_source)
 find_package(holoscan 1.0 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-# Fetch AJA nvt2 repository
-include(FetchContent)
-FetchContent_Declare(
-    ajantv2
-    GIT_REPOSITORY https://github.com/nvidia-holoscan/libajantv2.git
-    GIT_TAG d4250c556bcf1ebade627a3ef7a2027de7dc85ee
-)
-
-set(ENV{NTV2_VERSION_BUILD} 1)
-set(AJANTV2_DISABLE_DEMOS ON)
-set(AJANTV2_DISABLE_DRIVER ON)
-set(AJANTV2_DISABLE_PLUGINS ON)
-set(AJANTV2_DISABLE_TESTS ON)
-set(AJANTV2_DISABLE_TOOLS ON)
-set(AJA_INSTALL_HEADERS OFF)
-set(AJA_INSTALL_SOURCES OFF)
-message(STATUS "Fetching AJA sources")
-FetchContent_MakeAvailable(ajantv2)
-# Disable all the warnings for AJA
-target_compile_options(ajantv2 PRIVATE -w)
-
-add_library(AJA::ajantv2 ALIAS ajantv2)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_library(aja_source SHARED
   aja_source.hpp
-- 
2.34.1

