From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Mar 2023 17:02:57 -0700
Subject: Fix undefined fmt errors


diff --git a/operators/tensor_to_video_buffer/CMakeLists.txt b/operators/tensor_to_video_buffer/CMakeLists.txt
index 0c6de59..d9fca7a 100644
--- a/operators/tensor_to_video_buffer/CMakeLists.txt
+++ b/operators/tensor_to_video_buffer/CMakeLists.txt
@@ -31,6 +31,8 @@ target_link_libraries(
   holoscan::core
 )
 
+target_compile_definitions(tensor_to_video_buffer PUBLIC FMT_HEADER_ONLY)
+
 install(TARGETS tensor_to_video_buffer)
 
 if(HOLOHUB_BUILD_PYTHON)
diff --git a/operators/tool_tracking_postprocessor/CMakeLists.txt b/operators/tool_tracking_postprocessor/CMakeLists.txt
index a288d51..1e0ad0e 100644
--- a/operators/tool_tracking_postprocessor/CMakeLists.txt
+++ b/operators/tool_tracking_postprocessor/CMakeLists.txt
@@ -44,6 +44,8 @@ target_include_directories(tool_tracking_postprocessor
     ${CMAKE_CURRENT_SOURCE_DIR}
   )
 
+target_compile_definitions(tool_tracking_postprocessor PUBLIC FMT_HEADER_ONLY)
+
 if(HOLOHUB_BUILD_PYTHON)
     add_subdirectory(python)
 endif()
diff --git a/operators/visualizer_icardio/CMakeLists.txt b/operators/visualizer_icardio/CMakeLists.txt
index 01817d2..079fba5 100644
--- a/operators/visualizer_icardio/CMakeLists.txt
+++ b/operators/visualizer_icardio/CMakeLists.txt
@@ -35,6 +35,8 @@ target_include_directories(visualizer_icardio INTERFACE ${CMAKE_CURRENT_SOURCE_D
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MAJOR_VERSION=${holoscan_VERSION_MAJOR})
 target_compile_definitions(visualizer_icardio PRIVATE HOLOSCAN_MINOR_VERSION=${holoscan_VERSION_MINOR})
 
+target_compile_definitions(visualizer_icardio PUBLIC FMT_HEADER_ONLY)
+
 install(TARGETS visualizer_icardio)
 
 if(HOLOHUB_BUILD_PYTHON)
-- 
2.34.1

