From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Mar 2023 15:58:05 -0700
Subject: Fix ajantv2 dependencies


diff --git a/applications/endoscopy_out_of_body_detection/CMakeLists.txt b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
index e8e4fd0..f92e12b 100644
--- a/applications/endoscopy_out_of_body_detection/CMakeLists.txt
+++ b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
@@ -19,6 +19,8 @@ project(endoscopy_out_of_body_detection CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 option(HOLOHUB_DOWNLOAD_DATASETS "Download datasets" ON)
 
 add_executable(endoscopy_out_of_body_detection
diff --git a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
index cec5fcb..4e6da91 100644
--- a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
+++ b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(endoscopy_tool_tracking CXX)
 find_package(holoscan 0.5 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(endoscopy_tool_tracking
   main.cpp
 )
diff --git a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
index f0c93ea..7abf8a5 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multiai_endoscopy CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(multiai_endoscopy
   multi_ai.cpp
 )
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
index 87bcf3a..afeadd8 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
 enable_language(CUDA)
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
index 89a0f9f..9762a7b 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
@@ -20,6 +20,8 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
 enable_language(CUDA)
diff --git a/applications/multiai_ultrasound/cpp/CMakeLists.txt b/applications/multiai_ultrasound/cpp/CMakeLists.txt
index 8d845ee..95e479b 100644
--- a/applications/multiai_ultrasound/cpp/CMakeLists.txt
+++ b/applications/multiai_ultrasound/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(multiai_ultrasound CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(multiai_ultrasound
     main.cpp
 )
diff --git a/applications/object_detection_torch/CMakeLists.txt b/applications/object_detection_torch/CMakeLists.txt
index 8336cc8..6da18b0 100644
--- a/applications/object_detection_torch/CMakeLists.txt
+++ b/applications/object_detection_torch/CMakeLists.txt
@@ -19,6 +19,8 @@ project(object_detection_torch CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(object_detection_torch
     main.cpp
 )
diff --git a/applications/ultrasound_segmentation/cpp/CMakeLists.txt b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
index 97e8a4d..fa30504 100644
--- a/applications/ultrasound_segmentation/cpp/CMakeLists.txt
+++ b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
@@ -19,6 +19,8 @@ project(ultrasound_segmentation CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
+find_package(ajantv2 17.0 REQUIRED CONFIG)
+
 add_executable(ultrasound_segmentation
     main.cpp
 )
-- 
2.34.1

