From 0e2684fd239f38c18ce79fb126f5f40fc4e2caf0 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 9 Mar 2023 16:11:37 -0700
Subject: Build python libs with install RPATH


diff --git a/cmake/pybind11_add_holohub_module.cmake b/cmake/pybind11_add_holohub_module.cmake
index 2cc3aef..4a2690d 100644
--- a/cmake/pybind11_add_holohub_module.cmake
+++ b/cmake/pybind11_add_holohub_module.cmake
@@ -67,9 +67,8 @@ function(pybind11_add_holohub_module)
         "\$ORIGIN/../lib" # in our python wheel"
     )
     list(JOIN _rpath ":" _rpath)
-    set_property(TARGET ${target_name}
-        APPEND PROPERTY BUILD_RPATH ${_rpath}
-    )
+    set_target_properties(${target_name} PROPERTIES INSTALL_RPATH ${_rpath})
+    set_target_properties(${target_name} PROPERTIES BUILD_WITH_INSTALL_RPATH ON)
     unset(_rpath)
 
     # make submodule folder
-- 
2.34.1

