From dda8e39678197c7b9c90071949793085308bc0f1 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Thu, 23 Mar 2023 13:36:09 -0600
Subject: Fix default data paths in python apps


diff --git a/applications/colonoscopy_segmentation/colonoscopy_segmentation.py b/applications/colonoscopy_segmentation/colonoscopy_segmentation.py
index 1788c47..6e3b2a2 100644
--- a/applications/colonoscopy_segmentation/colonoscopy_segmentation.py
+++ b/applications/colonoscopy_segmentation/colonoscopy_segmentation.py
@@ -49,7 +49,7 @@ class ColonoscopyApp(Application):
         self.source = source
 
         if data == "none":
-            data = os.environ.get("HOLOSCAN_DATA_PATH", "../data")
+            data = os.environ.get("HOLOSCAN_DATA_PATH", "data/colonoscopy_segmentation")
 
         self.sample_data_path = data
 
diff --git a/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py b/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py
index e9d680e..eb93cdc 100644
--- a/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py
+++ b/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py
@@ -62,7 +62,7 @@ class EndoscopyApp(Application):
         self.source = source
 
         if data == "none":
-            data = os.environ.get("HOLOSCAN_DATA_PATH", "../data")
+            data = os.environ.get("HOLOSCAN_DATA_PATH", "data/endoscopy")
 
         self.sample_data_path = data
 
diff --git a/applications/multiai_ultrasound/python/multiai_ultrasound.py b/applications/multiai_ultrasound/python/multiai_ultrasound.py
index 53c9fde..eb2fe4c 100644
--- a/applications/multiai_ultrasound/python/multiai_ultrasound.py
+++ b/applications/multiai_ultrasound/python/multiai_ultrasound.py
@@ -35,7 +35,7 @@ class MultiAIICardio(Application):
         super().__init__()
 
         if data == "none":
-            data = os.environ.get("HOLOSCAN_DATA_PATH", "../data")
+            data = os.environ.get("HOLOSCAN_DATA_PATH", "data/multiai_ultrasound")
 
         # set name
         self.name = "Ultrasound App"
diff --git a/applications/ultrasound_segmentation/python/ultrasound_segmentation.py b/applications/ultrasound_segmentation/python/ultrasound_segmentation.py
index 80df4cd..231066e 100644
--- a/applications/ultrasound_segmentation/python/ultrasound_segmentation.py
+++ b/applications/ultrasound_segmentation/python/ultrasound_segmentation.py
@@ -49,7 +49,7 @@ class UltrasoundApp(Application):
         self.source = source
 
         if data == "none":
-            data = os.environ.get("HOLOSCAN_DATA_PATH", "../data")
+            data = os.environ.get("HOLOSCAN_DATA_PATH", "data/ultrasound_segmentation")
 
         self.sample_data_path = data
 
-- 
2.34.1

