From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 29 Nov 2023 17:00:19 -0700
Subject: Update AJA NTV2 dependency to 17.0


diff --git a/applications/endoscopy_out_of_body_detection/CMakeLists.txt b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
index a839c2f..f92e12b 100644
--- a/applications/endoscopy_out_of_body_detection/CMakeLists.txt
+++ b/applications/endoscopy_out_of_body_detection/CMakeLists.txt
@@ -19,7 +19,7 @@ project(endoscopy_out_of_body_detection CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 option(HOLOHUB_DOWNLOAD_DATASETS "Download datasets" ON)
 
diff --git a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
index 642079f..2469cdf 100644
--- a/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
+++ b/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt
@@ -19,7 +19,7 @@ project(endoscopy_tool_tracking CXX)
 find_package(holoscan 0.5 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_executable(endoscopy_tool_tracking
   main.cpp
diff --git a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
index 8ae6987..7abf8a5 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-cpu/CMakeLists.txt
@@ -20,7 +20,7 @@ project(multiai_endoscopy CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_executable(multiai_endoscopy
   multi_ai.cpp
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
index c2036b7..afeadd8 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-cpu/CMakeLists.txt
@@ -20,7 +20,7 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
diff --git a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
index c42eabe..9762a7b 100644
--- a/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
+++ b/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/CMakeLists.txt
@@ -20,7 +20,7 @@ project(multi_ai CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 # Enable cuda language
 set(CMAKE_CUDA_ARCHITECTURES "70;80")
diff --git a/applications/multiai_ultrasound/cpp/CMakeLists.txt b/applications/multiai_ultrasound/cpp/CMakeLists.txt
index b62ca61..95e479b 100644
--- a/applications/multiai_ultrasound/cpp/CMakeLists.txt
+++ b/applications/multiai_ultrasound/cpp/CMakeLists.txt
@@ -19,7 +19,7 @@ project(multiai_ultrasound CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_executable(multiai_ultrasound
     main.cpp
diff --git a/applications/object_detection_torch/CMakeLists.txt b/applications/object_detection_torch/CMakeLists.txt
index 496a819..05b51e0 100644
--- a/applications/object_detection_torch/CMakeLists.txt
+++ b/applications/object_detection_torch/CMakeLists.txt
@@ -19,7 +19,7 @@ project(object_detection_torch CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_executable(object_detection_torch
     main.cpp
diff --git a/applications/ultrasound_segmentation/cpp/CMakeLists.txt b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
index 62f2e3b..fa30504 100644
--- a/applications/ultrasound_segmentation/cpp/CMakeLists.txt
+++ b/applications/ultrasound_segmentation/cpp/CMakeLists.txt
@@ -19,7 +19,7 @@ project(ultrasound_segmentation CXX)
 find_package(holoscan 0.6 REQUIRED CONFIG
              PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
 
-find_package(ajantv2 16.2 REQUIRED CONFIG)
+find_package(ajantv2 17.0 REQUIRED CONFIG)
 
 add_executable(ultrasound_segmentation
     main.cpp
-- 
2.34.1

